--- wine.eselect	2019-06-01 07:44:56.930874433 +0200
+++ wine.eselect	2019-06-12 10:55:41.570175360 +0200
@@ -644,13 +644,6 @@
 }
 
 
-# lifted from fdo-mime.eclass
-desktop_database_update() {
-	if [[ -x "${EPREFIX}/usr/bin/update-desktop-database" ]]; then
-#		echo "Updating desktop mime database ..."
-		"${EPREFIX}/usr/bin/update-desktop-database" -q "${EPREFIX}/usr/share/applications" || die
-	fi
-}
 
 # sort function for versions, to be used in a pipe
 sort_versions() {
@@ -760,11 +753,6 @@
 	# Remove active if it exists
 	store_config "${CONFIG_DIR}/active" "${v}" ""
 
-	local desktop="${EPREFIX}/usr/share/applications/wine${v_arg#-}.desktop"
-	if [[ -e "${desktop}" ]]; then
-		rm "${desktop}" || die "Failed ro remove desktop file"
-	fi
-	desktop_database_update || die "Failed to update xorg desktop database"
 }
 
 set_symlinks() {
@@ -872,10 +860,6 @@
 	store_config "${CONFIG_DIR}/links/${v}" "include" "${link}"
 
 
-	local desktop="${EPREFIX}/usr/share/applications/wine${v_arg#-}.desktop"
-	cp "${wine_app}/wine.desktop" "${desktop}" || die "Failed to copy desktop file"
-	sed -i "s:Exec=.*:Exec=wine${v_arg#-} start /unix %f:" ${desktop} || die "Failed to edit desktop file"
-	desktop_database_update || die "Failed to update xorg desktop database"
 
 	)
 	unset_dir_env
