diff --git a/src/config.cpp b/src/config.cpp
index 58b8ce9..e9e53d1 100644
--- a/src/config.cpp
+++ b/src/config.cpp
@@ -61,17 +61,30 @@ std::string get_program_name() {
     return basename;
 }
 
+static std::string get_gpu_vendor()
+{
+    if (const auto glx_vendor = getenv("__GLX_VENDOR_LIBRARY_NAME"))
+    {
+        return std::string(glx_vendor);
+    }
+    else
+    {
+        return "mesa";
+    }
+}
+
 static void enumerate_config_files(std::vector<std::string>& paths) {
     static const char *mangohud_dir = "/MangoHud/";
 
     const std::string config_dir = get_config_dir();
+    const std::string gpu_vendor = get_gpu_vendor();
 
     if (config_dir.empty()) {
         // If we can't find 'HOME' just abandon hope.
         return;
     }
 
-    paths.push_back(config_dir + mangohud_dir + "MangoHud.conf");
+    paths.push_back(config_dir + mangohud_dir + "MangoHud-" + gpu_vendor + ".conf");
 }
 
 bool parseConfigFile(overlay_params& params) {
