diff --git a/Makefile b/Makefile
index 5fcf1dd..3c054c3 100644
--- a/Makefile
+++ b/Makefile
@@ -253,7 +253,7 @@ IP6=6
 #	British.  As of 2005, the Julian calendar and the Gregorian calendar
 #	diverge by 15 days.
 
-EXTRACFLAGS=
+EXTRACFLAGS=-std=gnu99
 
 
 # Extra linker flags (additional/alternative libraries, etc.)
@@ -699,7 +699,7 @@ rebuild:
 	@echo Rebuilding c-client for `$(CAT) OSTYPE`...
 	@$(TOUCH) SPECIALS
 	$(CD) c-client;$(MAKE) all CC=`$(CAT) CCTYPE` \
-	 CFLAGS="`$(CAT) CFLAGS`" `$(CAT) SPECIALS`
+	 CFLAGS="`$(CAT) CFLAGS` -std=gnu99" `$(CAT) SPECIALS`
 
 rebuildclean:
 	$(SH) -c '$(RM) rebuild || true'
diff --git a/src/dmail/Makefile b/src/dmail/Makefile
index 4aef24c..572278e 100644
--- a/src/dmail/Makefile
+++ b/src/dmail/Makefile
@@ -32,7 +32,7 @@ SHELL = /bin/sh
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
-CFLAGS = -I$C `cat $C/CFLAGS`
+CFLAGS = -I$C `cat $C/CFLAGS` -std=gnu99
 LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
 
 dmail: $(CCLIENTLIB) dmail.o dquota.o
diff --git a/src/ipopd/Makefile b/src/ipopd/Makefile
index accb540..c8b75c0 100644
--- a/src/ipopd/Makefile
+++ b/src/ipopd/Makefile
@@ -32,7 +32,7 @@ SHELL = /bin/sh
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
-CFLAGS = -I$C `cat $C/CFLAGS`
+CFLAGS = -I$C `cat $C/CFLAGS` -std=gnu99
 LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
 
 ipopd: ipop2d ipop3d
diff --git a/src/mailutil/Makefile b/src/mailutil/Makefile
index acc58bb..af49b36 100644
--- a/src/mailutil/Makefile
+++ b/src/mailutil/Makefile
@@ -32,7 +32,7 @@ SHELL = /bin/sh
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
-CFLAGS = -I$C `cat $C/CFLAGS`
+CFLAGS = -I$C `cat $C/CFLAGS` -std=gnu99
 LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
 
 mailutil: $(CCLIENTLIB) mailutil.o
diff --git a/src/mlock/Makefile b/src/mlock/Makefile
index b7dc44c..2abc04b 100644
--- a/src/mlock/Makefile
+++ b/src/mlock/Makefile
@@ -31,7 +31,7 @@ SHELL = /bin/sh
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
-CFLAGS = `cat $C/CFLAGS`
+CFLAGS = `cat $C/CFLAGS` -std=gnu99
 
 all:	mlock
 
diff --git a/src/mtest/Makefile b/src/mtest/Makefile
index a49be58..9650b1f 100644
--- a/src/mtest/Makefile
+++ b/src/mtest/Makefile
@@ -32,7 +32,7 @@ SHELL = /bin/sh
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
-CFLAGS = -I$C `cat $C/CFLAGS`
+CFLAGS = -I$C `cat $C/CFLAGS` -std=gnu99
 LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
 
 all:	mtest
diff --git a/src/osdep/amiga/mbx.c b/src/osdep/amiga/mbx.c
index 1ece5d8..f59895a 100644
--- a/src/osdep/amiga/mbx.c
+++ b/src/osdep/amiga/mbx.c
@@ -34,6 +34,8 @@
  * The ctime is the last write time of the file.
  */
 
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -43,6 +45,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/amiga/mh.c b/src/osdep/amiga/mh.c
index 0226b7a..e7c907a 100644
--- a/src/osdep/amiga/mh.c
+++ b/src/osdep/amiga/mh.c
@@ -36,6 +36,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/amiga/mix.c b/src/osdep/amiga/mix.c
index fbf4a02..89750f6 100644
--- a/src/osdep/amiga/mix.c
+++ b/src/osdep/amiga/mix.c
@@ -24,6 +24,7 @@
  * Last Edited:	7 May 2008
  */
 
+#define _GNU_SOURCE
 
 #include <stdio.h>
 #include <ctype.h>
@@ -34,6 +35,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/amiga/mmdf.c b/src/osdep/amiga/mmdf.c
index e962434..c7bd6a7 100644
--- a/src/osdep/amiga/mmdf.c
+++ b/src/osdep/amiga/mmdf.c
@@ -24,6 +24,7 @@
  * Last Edited:	27 March 2008
  */
 
+#define _GNU_SOURCE
 
 #include <stdio.h>
 #include <ctype.h>
@@ -33,6 +34,7 @@ extern int errno;		/* just in case */
 #include "mail.h"
 #include "osdep.h"
 #include <time.h>
+#include <utime.h>
 #include <sys/stat.h>
 #include "pseudo.h"
 #include "fdstring.h"
diff --git a/src/osdep/amiga/mtx.c b/src/osdep/amiga/mtx.c
index 8e6f76e..4dc7da4 100644
--- a/src/osdep/amiga/mtx.c
+++ b/src/osdep/amiga/mtx.c
@@ -34,6 +34,8 @@
  * The ctime is the last write time of the file.
  */
 
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -43,6 +45,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/amiga/mx.c b/src/osdep/amiga/mx.c
index 4549527..9281a35 100644
--- a/src/osdep/amiga/mx.c
+++ b/src/osdep/amiga/mx.c
@@ -26,6 +26,7 @@
  * Last Edited:	6 January 2008
  */
 
+#define _GNU_SOURCE
 
 #include <stdio.h>
 #include <ctype.h>
@@ -36,6 +37,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/amiga/tenex.c b/src/osdep/amiga/tenex.c
index eee61fb..9b5029d 100644
--- a/src/osdep/amiga/tenex.c
+++ b/src/osdep/amiga/tenex.c
@@ -39,6 +39,8 @@
  * msg.text size.  Beware.
  */
 
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -48,6 +50,7 @@ extern int errno;		/* just in case */
 #include <sys/stat.h>
 #include "misc.h"
 #include "dummy.h"
+#include <utime.h>
 
 /* TENEX I/O stream local data */
 	
diff --git a/src/osdep/amiga/unix.c b/src/osdep/amiga/unix.c
index be3c437..1245e07 100644
--- a/src/osdep/amiga/unix.c
+++ b/src/osdep/amiga/unix.c
@@ -37,6 +37,13 @@
  *  Lift a leg, Yunie.  Luv ya forever!!!!
  */
 
+#define _GNU_SOURCE
+#define _XOPEN_SOURCE
+#define __USE_XOPEN_EXTENDED
+#define _POSIX_C_SOURCE
+#define __USE_XOPEN2K
+#define __USE_XOPEN
+
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -45,6 +52,7 @@ extern int errno;		/* just in case */
 #include "mail.h"
 #include "osdep.h"
 #include <time.h>
+#include <utime.h>
 #include <sys/stat.h>
 #include "unix.h"
 #include "pseudo.h"
diff --git a/src/osdep/amiga/unix.h b/src/osdep/amiga/unix.h
index 76cde8b..789f07c 100644
--- a/src/osdep/amiga/unix.h
+++ b/src/osdep/amiga/unix.h
@@ -47,6 +47,13 @@
  * Returns: t,ti,zn set if valid From string, else ti is NIL
  */
 
+#define _GNU_SOURCE
+#define _XOPEN_SOURCE
+#define __USE_XOPEN_EXTENDED
+#define _POSIX_C_SOURCE
+#define __USE_XOPEN2K
+#define __USE_XOPEN
+
 #define VALID(s,x,ti,zn) {						\
   int remote = 0;							\
   ti = 0;								\
diff --git a/src/osdep/unix/Makefile b/src/osdep/unix/Makefile
index 78913ac..233844a 100644
--- a/src/osdep/unix/Makefile
+++ b/src/osdep/unix/Makefile
@@ -48,8 +48,8 @@ GCCOPTLEVEL= -O2
 # Try to have some consistency in GCC builds.  We want optimization, but we
 # also want to be able to debug.
 
-GCCCFLAGS= -g $(GCCOPTLEVEL) -pipe -fno-omit-frame-pointer
-GCC4CFLAGS= $(GCCCFLAGS) -Wno-pointer-sign
+GCCCFLAGS= -g $(GCCOPTLEVEL) -pipe -fno-omit-frame-pointer -std=gnu99
+GCC4CFLAGS= $(GCCCFLAGS) -Wno-pointer-sign -std=gnu99
 
 
 # Extended flags needed for SSL.  You may need to modify.
@@ -79,7 +79,7 @@ SSLLDFLAGS= -L$(SSLLIB) -lssl $(SSLCRYPTO) $(SSLRSA)
 # Extended flags needed for non-standard passwd types.  You may need to modify.
 
 AFSDIR=/usr/afsws
-AFSCFLAGS=-I$(AFSDIR)/include
+AFSCFLAGS=-I$(AFSDIR)/include -std=gnu99
 AFSLIB=$(AFSDIR)/lib
 AFSLDFLAGS=-L$(AFSLIB)/afs -L$(AFSLIB) -L$(AFSDIR)/domestic/lib\
  -lkauth -lprot -lubik -lauth -lrxkad -lrx -llwp -ldes -lcom_err\
@@ -154,7 +154,7 @@ BINARIES=osdep.o mail.o misc.o newsrc.o smanager.o utf8.o utf8aux.o siglocal.o \
  dummy.o pseudo.o netmsg.o flstring.o fdstring.o \
  rfc822.o nntp.o smtp.o imap4r1.o pop3.o \
  unix.o mbx.o mmdf.o tenex.o mtx.o news.o phile.o mh.o mx.o mix.o
-CFLAGS=-g
+CFLAGS=-g -std=gnu99
 
 CAT=cat
 MAKE=make
diff --git a/src/osdep/unix/env_unix.h b/src/osdep/unix/env_unix.h
index 365e512..a29191a 100644
--- a/src/osdep/unix/env_unix.h
+++ b/src/osdep/unix/env_unix.h
@@ -26,6 +26,7 @@
  * Last Edited:	30 August 2006
  */
 
+#define _GNU_SOURCE
 
 typedef struct dotlock_base {
   char lock[MAILTMPLEN];
diff --git a/src/osdep/unix/mbx.c b/src/osdep/unix/mbx.c
index 1ece5d8..f59895a 100644
--- a/src/osdep/unix/mbx.c
+++ b/src/osdep/unix/mbx.c
@@ -34,6 +34,8 @@
  * The ctime is the last write time of the file.
  */
 
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -43,6 +45,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/unix/mh.c b/src/osdep/unix/mh.c
index 0226b7a..e7c907a 100644
--- a/src/osdep/unix/mh.c
+++ b/src/osdep/unix/mh.c
@@ -36,6 +36,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/unix/mix.c b/src/osdep/unix/mix.c
index fbf4a02..89750f6 100644
--- a/src/osdep/unix/mix.c
+++ b/src/osdep/unix/mix.c
@@ -24,6 +24,7 @@
  * Last Edited:	7 May 2008
  */
 
+#define _GNU_SOURCE
 
 #include <stdio.h>
 #include <ctype.h>
@@ -34,6 +35,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/unix/mmdf.c b/src/osdep/unix/mmdf.c
index e962434..c7bd6a7 100644
--- a/src/osdep/unix/mmdf.c
+++ b/src/osdep/unix/mmdf.c
@@ -24,6 +24,7 @@
  * Last Edited:	27 March 2008
  */
 
+#define _GNU_SOURCE
 
 #include <stdio.h>
 #include <ctype.h>
@@ -33,6 +34,7 @@ extern int errno;		/* just in case */
 #include "mail.h"
 #include "osdep.h"
 #include <time.h>
+#include <utime.h>
 #include <sys/stat.h>
 #include "pseudo.h"
 #include "fdstring.h"
diff --git a/src/osdep/unix/mtx.c b/src/osdep/unix/mtx.c
index 8e6f76e..4dc7da4 100644
--- a/src/osdep/unix/mtx.c
+++ b/src/osdep/unix/mtx.c
@@ -34,6 +34,8 @@
  * The ctime is the last write time of the file.
  */
 
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -43,6 +45,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/unix/mx.c b/src/osdep/unix/mx.c
index 4549527..9281a35 100644
--- a/src/osdep/unix/mx.c
+++ b/src/osdep/unix/mx.c
@@ -26,6 +26,7 @@
  * Last Edited:	6 January 2008
  */
 
+#define _GNU_SOURCE
 
 #include <stdio.h>
 #include <ctype.h>
@@ -36,6 +37,7 @@ extern int errno;		/* just in case */
 #include <pwd.h>
 #include <sys/stat.h>
 #include <sys/time.h>
+#include <utime.h>
 #include "misc.h"
 #include "dummy.h"
 #include "fdstring.h"
diff --git a/src/osdep/unix/os_lnx.h b/src/osdep/unix/os_lnx.h
index b5f39ff..3b1bb3a 100644
--- a/src/osdep/unix/os_lnx.h
+++ b/src/osdep/unix/os_lnx.h
@@ -59,6 +59,7 @@
 
 #define flock safe_flock
 
+extern int safe_flock (int fd,int op);
 
 #include "env_unix.h"
 #include "fs.h"
diff --git a/src/osdep/unix/os_slx.h b/src/osdep/unix/os_slx.h
index b5f39ff..3b1bb3a 100644
--- a/src/osdep/unix/os_slx.h
+++ b/src/osdep/unix/os_slx.h
@@ -59,6 +59,7 @@
 
 #define flock safe_flock
 
+extern int safe_flock (int fd,int op);
 
 #include "env_unix.h"
 #include "fs.h"
diff --git a/src/osdep/unix/tenex.c b/src/osdep/unix/tenex.c
index eee61fb..9b5029d 100644
--- a/src/osdep/unix/tenex.c
+++ b/src/osdep/unix/tenex.c
@@ -39,6 +39,8 @@
  * msg.text size.  Beware.
  */
 
+#define _GNU_SOURCE
+
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -48,6 +50,7 @@ extern int errno;		/* just in case */
 #include <sys/stat.h>
 #include "misc.h"
 #include "dummy.h"
+#include <utime.h>
 
 /* TENEX I/O stream local data */
 	
diff --git a/src/osdep/unix/unix.c b/src/osdep/unix/unix.c
index be3c437..1245e07 100644
--- a/src/osdep/unix/unix.c
+++ b/src/osdep/unix/unix.c
@@ -37,6 +37,13 @@
  *  Lift a leg, Yunie.  Luv ya forever!!!!
  */
 
+#define _GNU_SOURCE
+#define _XOPEN_SOURCE
+#define __USE_XOPEN_EXTENDED
+#define _POSIX_C_SOURCE
+#define __USE_XOPEN2K
+#define __USE_XOPEN
+
 #include <stdio.h>
 #include <ctype.h>
 #include <errno.h>
@@ -45,6 +52,7 @@ extern int errno;		/* just in case */
 #include "mail.h"
 #include "osdep.h"
 #include <time.h>
+#include <utime.h>
 #include <sys/stat.h>
 #include "unix.h"
 #include "pseudo.h"
diff --git a/src/osdep/unix/unix.h b/src/osdep/unix/unix.h
index 1305184..7b52f3b 100644
--- a/src/osdep/unix/unix.h
+++ b/src/osdep/unix/unix.h
@@ -47,6 +47,13 @@
  * Returns: t,ti,zn set if valid From string, else ti is NIL
  */
 
+#define _GNU_SOURCE
+#define _XOPEN_SOURCE
+#define __USE_XOPEN_EXTENDED
+#define _POSIX_C_SOURCE
+#define __USE_XOPEN2K
+#define __USE_XOPEN
+
 #define VALID(s,x,ti,zn) {						\
   ti = 0;								\
   if ((*s == 'F') && (s[1] == 'r') && (s[2] == 'o') && (s[3] == 'm') &&	\
diff --git a/src/tmail/Makefile b/src/tmail/Makefile
index ce22143..cf10b14 100644
--- a/src/tmail/Makefile
+++ b/src/tmail/Makefile
@@ -32,7 +32,7 @@ SHELL = /bin/sh
 # Get local definitions from c-client directory
 
 CC = `cat $C/CCTYPE`
-CFLAGS = -I$C `cat $C/CFLAGS`
+CFLAGS = -I$C `cat $C/CFLAGS` -std=gnu99
 LDFLAGS = $(CCLIENTLIB) `cat $C/LDFLAGS`
 
 tmail: $(CCLIENTLIB) tmail.o tquota.o
diff --git a/tools/Makefile b/tools/Makefile
index f2db51f..878de65 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -26,7 +26,7 @@ RM=rm -f
 
 
 uahelper:
-	$(CC) -o uahelper uahelper.c $(LDFLAGS)
+	$(CC) -o uahelper -std=gnu99 uahelper.c $(LDFLAGS)
 
 clean:
 	sh -c '$(RM) *.o uahelper || true'
