diff --git a/src/net/cgo_linux.go b/src/net/cgo_linux.go
index de6e87f..8a0872f 100644
--- a/src/net/cgo_linux.go
+++ b/src/net/cgo_linux.go
@@ -7,6 +7,7 @@
 package net
 
 /*
+#cgo linux CFLAGS: -D_GNU_SOURCE
 #include <netdb.h>
 */
 import "C"
diff --git a/src/os/user/getgrouplist_unix.go b/src/os/user/getgrouplist_unix.go
index fb482d3..66fbee5 100644
--- a/src/os/user/getgrouplist_unix.go
+++ b/src/os/user/getgrouplist_unix.go
@@ -7,6 +7,7 @@
 package user
 
 /*
+#cgo CFLAGS: -D_GNU_SOURCE
 #include <unistd.h>
 #include <sys/types.h>
 #include <grp.h>
diff --git a/src/plugin/plugin_dlopen.go b/src/plugin/plugin_dlopen.go
index f6ae219..69d5e33 100644
--- a/src/plugin/plugin_dlopen.go
+++ b/src/plugin/plugin_dlopen.go
@@ -7,6 +7,7 @@
 package plugin
 
 /*
+#cgo linux CFLAGS: -D_GNU_SOURCE
 #cgo linux LDFLAGS: -ldl
 #include <dlfcn.h>
 #include <limits.h>
diff --git a/src/runtime/cgo/gcc_setenv.c b/src/runtime/cgo/gcc_setenv.c
index 47caa4b..5a96b2f 100644
--- a/src/runtime/cgo/gcc_setenv.c
+++ b/src/runtime/cgo/gcc_setenv.c
@@ -4,6 +4,8 @@
 
 //go:build unix
 
+#define _GNU_SOURCE
+
 #include "libcgo.h"
 
 #include <stdlib.h>
diff --git a/src/runtime/cgo/gcc_sigaction.c b/src/runtime/cgo/gcc_sigaction.c
index 374909b..dffd47d 100644
--- a/src/runtime/cgo/gcc_sigaction.c
+++ b/src/runtime/cgo/gcc_sigaction.c
@@ -4,6 +4,8 @@
 
 //go:build linux && (amd64 || arm64 || ppc64le)
 
+#define _GNU_SOURCE
+
 #include <errno.h>
 #include <stddef.h>
 #include <stdint.h>
diff --git a/src/runtime/cgo/libcgo.h b/src/runtime/cgo/libcgo.h
index 04755f0..a892e31 100644
--- a/src/runtime/cgo/libcgo.h
+++ b/src/runtime/cgo/libcgo.h
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+#define _GNU_SOURCE
+
 #include <stdint.h>
 #include <stdlib.h>
 #include <stdio.h>
