diff --git a/src/runtime/cgo/cgo.go b/src/runtime/cgo/cgo.go
index 1e3a502..0f82fdf 100644
--- a/src/runtime/cgo/cgo.go
+++ b/src/runtime/cgo/cgo.go
@@ -26,7 +26,7 @@ package cgo
 // Use -fno-stack-protector to avoid problems locating the
 // proper support functions. See issues #52919, #54313, #58385.
 // Use -Wdeclaration-after-statement because some CI builds use it.
-#cgo CFLAGS: -Wall -Werror -fno-stack-protector -Wdeclaration-after-statement
+#cgo CFLAGS: -Wall -fno-stack-protector -Wdeclaration-after-statement
 
 #cgo solaris CPPFLAGS: -D_POSIX_PTHREAD_SEMANTICS
 
