diff --git a/src/crypto/internal/sysrand/internal/seccomp/seccomp_linux.go b/src/crypto/internal/sysrand/internal/seccomp/seccomp_linux.go
index 32ef52a..e6b6ec0 100644
--- a/src/crypto/internal/sysrand/internal/seccomp/seccomp_linux.go
+++ b/src/crypto/internal/sysrand/internal/seccomp/seccomp_linux.go
@@ -5,6 +5,8 @@
 package seccomp
 
 /*
+#cgo CFLAGS: -D_GNU_SOURCE
+
 #include <sys/prctl.h>
 #include <sys/syscall.h>
 #include <errno.h>
