diff --git a/app/systemsettings.desktop b/app/systemsettings.desktop
index 4d64b88..5bc3fbe 100755
--- a/app/systemsettings.desktop
+++ b/app/systemsettings.desktop
@@ -357,7 +357,7 @@ Name[x-test]=xxGlobal Themexx
 Name[zh_CN]=全局主题
 Name[zh_TW]=全域主題
 Icon=preferences-desktop-theme-global
-Exec=systemsettings kcm_lookandfeel
+Exec=kcmshell6 kcm_lookandfeel
 
 [Desktop Action kcm-users]
 Name=Users
@@ -418,7 +418,7 @@ Name[x-test]=xxUsersxx
 Name[zh_CN]=用户
 Name[zh_TW]=使用者
 Icon=preferences-system-users
-Exec=systemsettings kcm_users
+Exec=kcmshell6 kcm_users
 
 [Desktop Action kcm-screenlocker]
 Name=Screen Locking
@@ -478,7 +478,7 @@ Name[x-test]=xxScreen Lockingxx
 Name[zh_CN]=锁屏
 Name[zh_TW]=螢幕鎖定
 Icon=preferences-desktop-user-password
-Exec=systemsettings kcm_screenlocker
+Exec=kcmshell6 kcm_screenlocker
 
 [Desktop Action kcm-powerdevilprofilesconfig]
 Name=Power Management
@@ -564,7 +564,7 @@ Name[x-test]=xxPower Managementxx
 Name[zh_CN]=电源管理
 Name[zh_TW]=電源管理
 Icon=preferences-system-power-management
-Exec=systemsettings kcm_powerdevilprofilesconfig
+Exec=kcmshell6 kcm_powerdevilprofilesconfig
 
 [Desktop Action kcm-kscreen]
 Name=Display Configuration
@@ -623,4 +623,4 @@ Name[x-test]=xxDisplay Configurationxx
 Name[zh_CN]=显示器配置
 Name[zh_TW]=顯示器設定
 Icon=preferences-desktop-display-randr
-Exec=systemsettings kcm_kscreen
+Exec=kcmshell6 kcm_kscreen
