diff --git a/src/core/guesslanguage.cpp b/src/core/guesslanguage.cpp
index 152a399..f5d752a 100644
--- a/src/core/guesslanguage.cpp
+++ b/src/core/guesslanguage.cpp
@@ -649,9 +649,6 @@ void GuessLanguagePrivate::loadModels()
     }
     QSet<QString> knownLanguages(s_scriptLanguages.constBegin(), s_scriptLanguages.constEnd());
     knownLanguages.subtract(availableLanguages);
-    if (!knownLanguages.isEmpty()) {
-        qCDebug(SONNET_LOG_CORE) << "Missing trigrams for languages:" << knownLanguages;
-    }
 }
 
 QList<QChar::Script> GuessLanguagePrivate::findRuns(const QString &text)
