diff --git a/smb/transfer.h b/smb/transfer.h
index dace313..14ef0aa 100644
--- a/smb/transfer.h
+++ b/smb/transfer.h
@@ -12,6 +12,8 @@
 #include <array>
 #include <condition_variable>
 #include <mutex>
+#include <array>
+#include <memory>
 
 constexpr off_t c_minSegmentSize = 64 * 1024; // minimal size on stack
 constexpr off_t c_maxSegmentSize = 4L * 1024 * 1024; // 4MiB is the largest request we make
