diff --git a/resources/google-groupware/calendarhandler.cpp b/resources/google-groupware/calendarhandler.cpp
index e0edfc1..5dbe292 100644
--- a/resources/google-groupware/calendarhandler.cpp
+++ b/resources/google-groupware/calendarhandler.cpp
@@ -381,7 +381,7 @@ void FreeBusyHandler::retrieveFreeBusy(const QString &email, const QDateTime &st
         }
 
         KCalendarCore::FreeBusy::Ptr fb(new KCalendarCore::FreeBusy);
-        fb->setUid(QStringLiteral("%1%2@google.com").arg(QDateTime::currentDateTimeUtc().toString(QStringLiteral("yyyyMMddTHHmmssZ"))));
+        fb->setUid(QStringLiteral("%1%2@nothanks.invalid").arg(QDateTime::currentDateTimeUtc().toString(QStringLiteral("yyyyMMddTHHmmssZ"))));
         fb->setOrganizer(job->account()->accountName());
         fb->addAttendee(KCalendarCore::Attendee(QString(), queryJob->id()));
         // FIXME: is it really sort?
diff --git a/resources/google-groupware/contacthandler.cpp b/resources/google-groupware/contacthandler.cpp
index a56b42b..a7ba5f9 100644
--- a/resources/google-groupware/contacthandler.cpp
+++ b/resources/google-groupware/contacthandler.cpp
@@ -69,7 +69,7 @@ bool ContactHandler::canPerformTask(const Item::List &items)
 
 QString ContactHandler::myContactsRemoteId() const
 {
-    return QStringLiteral("http://www.google.com/m8/feeds/groups/%1/base/6").arg(QString::fromLatin1(QUrl::toPercentEncoding(m_settings->accountPtr()->accountName())));
+    return {};
 }
 
 void ContactHandler::setupCollection(Collection &collection, const ContactsGroupPtr &group)
diff --git a/resources/google-groupware/googleresource.cpp b/resources/google-groupware/googleresource.cpp
index 8c2af27..281492c 100644
--- a/resources/google-groupware/googleresource.cpp
+++ b/resources/google-groupware/googleresource.cpp
@@ -138,7 +138,6 @@ void GoogleResource::configure(WId windowId)
         Q_EMIT configurationDialogAccepted();
 
         if (m_settings->accountPtr().isNull()) {
-            Q_EMIT status(NotConfigured, i18n("Configured account does not exist"));
             m_isConfiguring = false;
             return;
         }
@@ -164,18 +163,10 @@ QList<QUrl> GoogleResource::scopes() const
 
 void GoogleResource::updateResourceName()
 {
-    const QString accountName = m_settings->account();
-    setName(i18nc("%1 is account name (user@gmail.com)", "Google Groupware (%1)", accountName.isEmpty() ? i18n("not configured") : accountName));
 }
 
 void GoogleResource::reloadConfig()
 {
-    const AccountPtr account = m_settings->accountPtr();
-    if (account.isNull() || account->accountName().isEmpty()) {
-        Q_EMIT status(NotConfigured, i18n("Configured account does not exist"));
-    } else {
-        emitReadyStatus();
-    }
 }
 
 bool GoogleResource::handleError(KGAPI2::Job *job, bool _cancelTask)
@@ -207,13 +198,7 @@ bool GoogleResource::handleError(KGAPI2::Job *job, bool _cancelTask)
 
 bool GoogleResource::canPerformTask()
 {
-    if (!m_settings->accountPtr() && accountId() == 0) {
-        cancelTask(i18nc("@info:status", "Resource is not configured"));
-        Q_EMIT status(NotConfigured, i18nc("@info:status", "Resource is not configured"));
-        return false;
-    }
-
-    return true;
+    return false;
 }
 
 void GoogleResource::slotAuthJobFinished(KGAPI2::Job *job)
diff --git a/resources/google-groupware/googlesettings.cpp b/resources/google-groupware/googlesettings.cpp
index f07950b..17d4b5c 100644
--- a/resources/google-groupware/googlesettings.cpp
+++ b/resources/google-groupware/googlesettings.cpp
@@ -153,17 +153,17 @@ void GoogleSettings::addTaskList(const QString &taskList)
 
 QString GoogleSettings::clientId() const
 {
-    return QStringLiteral("554041944266.apps.googleusercontent.com");
+    return {};
 }
 
 QString GoogleSettings::clientSecret() const
 {
-    return QStringLiteral("mdT1DjzohxN3npUUzkENT0gO");
+    return {};
 }
 
 bool GoogleSettings::isReady() const
 {
-    return m_isReady;
+    return false;
 }
 
 AccountPtr GoogleSettings::accountPtr()
diff --git a/resources/imap/gmailpasswordrequester.cpp b/resources/imap/gmailpasswordrequester.cpp
index dd7baed..7761f57 100644
--- a/resources/imap/gmailpasswordrequester.cpp
+++ b/resources/imap/gmailpasswordrequester.cpp
@@ -26,8 +26,8 @@
 #include <KGAPI/AuthJob>
 #include <KGAPI/AccountManager>
 
-#define GOOGLE_API_KEY QStringLiteral("554041944266.apps.googleusercontent.com")
-#define GOOGLE_API_SECRET QStringLiteral("mdT1DjzohxN3npUUzkENT0gO")
+#define GOOGLE_API_KEY QStringLiteral("")
+#define GOOGLE_API_SECRET QStringLiteral("")
 
 GmailPasswordRequester::GmailPasswordRequester(ImapResourceBase *resource, QObject *parent)
     : PasswordRequesterInterface(parent)
