diff --git a/agents/migration/CMakeLists.txt b/agents/migration/CMakeLists.txt
index 31f6bc4..3c7c88b 100644
--- a/agents/migration/CMakeLists.txt
+++ b/agents/migration/CMakeLists.txt
@@ -34,7 +34,7 @@ endif()
 
 target_link_libraries(akonadi_migration_agent
     gidmigration
-    googlegroupwaremigration
+    #googlegroupwaremigration
     icalcategoriestotagsmigration
     KPim6::AkonadiCore
     KPim6::AkonadiAgentBase
diff --git a/agents/migration/migrationagent.cpp b/agents/migration/migrationagent.cpp
index 32fcb08..2ea5a37 100644
--- a/agents/migration/migrationagent.cpp
+++ b/agents/migration/migrationagent.cpp
@@ -9,7 +9,6 @@
 #include "migrationstatuswidget.h"
 
 #include "migration/gid/gidmigrator.h"
-#include "migration/googlegroupware/googleresourcemigrator.h"
 #include "migration/icalcategoriestotags/icalcategoriestotagsmigrator.h"
 
 #include <KContacts/Addressee>
@@ -28,7 +27,6 @@ MigrationAgent::MigrationAgent(const QString &id)
 {
     KLocalizedString::setApplicationDomain(QByteArrayLiteral("akonadi_migration_agent"));
     mScheduler.addMigrator(QSharedPointer<GidMigrator>::create(KContacts::Addressee::mimeType()));
-    mScheduler.addMigrator(QSharedPointer<GoogleResourceMigrator>::create());
     mScheduler.addMigrator(QSharedPointer<ICalCategoriesToTagsMigrator>::create());
 }
 
diff --git a/migration/CMakeLists.txt b/migration/CMakeLists.txt
index 6a74671..b9284d4 100644
--- a/migration/CMakeLists.txt
+++ b/migration/CMakeLists.txt
@@ -36,6 +36,6 @@ target_link_libraries(migrationshared
 
 
 add_subdirectory(gid)
-add_subdirectory(googlegroupware)
+#add_subdirectory(googlegroupware)
 add_subdirectory(icalcategoriestotags)
 
diff --git a/resources/CMakeLists.txt b/resources/CMakeLists.txt
index eca9216..0f79740 100644
--- a/resources/CMakeLists.txt
+++ b/resources/CMakeLists.txt
@@ -57,7 +57,7 @@ add_subdirectory(maildir)
 add_subdirectory(openxchange)
 add_subdirectory(pop3)
 
-add_subdirectory(google-groupware)
+#add_subdirectory(google-groupware)
 
 add_subdirectory(shared)
 add_subdirectory(birthdays)
diff --git a/resources/google-groupware/calendarhandler.cpp b/resources/google-groupware/calendarhandler.cpp
index 1da8c14..0427522 100644
--- a/resources/google-groupware/calendarhandler.cpp
+++ b/resources/google-groupware/calendarhandler.cpp
@@ -451,7 +451,7 @@ void FreeBusyHandler::retrieveFreeBusy(const QString &email, const QDateTime &st
         }
 
         KCalendarCore::FreeBusy::Ptr fb(new KCalendarCore::FreeBusy);
-        fb->setUid(QStringLiteral("%1%2@google.com").arg(QDateTime::currentDateTimeUtc().toString(QStringLiteral("yyyyMMddTHHmmssZ"))));
+        fb->setUid(QStringLiteral("%1%2@nothanks.invalid").arg(QDateTime::currentDateTimeUtc().toString(QStringLiteral("yyyyMMddTHHmmssZ"))));
         fb->setOrganizer(job->account()->accountName());
         fb->addAttendee(KCalendarCore::Attendee(QString(), queryJob->id()));
         // FIXME: is it really sort?
diff --git a/resources/google-groupware/googleresource.cpp b/resources/google-groupware/googleresource.cpp
index 6e4cf37..ea2c0af 100644
--- a/resources/google-groupware/googleresource.cpp
+++ b/resources/google-groupware/googleresource.cpp
@@ -142,16 +142,10 @@ QList<QUrl> GoogleResource::scopes() const
 
 void GoogleResource::updateResourceName()
 {
-    const QString accountName = m_settings.account();
-    setName(i18nc("%1 is account name (user@gmail.com)", "Google Groupware (%1)", accountName.isEmpty() ? i18n("Not configured") : accountName));
 }
 
 void GoogleResource::reloadConfig()
 {
-    m_settings.load();
-    m_settings.init();
-
-    updateResourceName();
 }
 
 bool GoogleResource::handleError(KGAPI2::Job *job, bool _cancelTask)
@@ -218,13 +212,7 @@ void GoogleResource::requestAuthenticationFromUser(const KGAPI2::AccountPtr &acc
 
 bool GoogleResource::canPerformTask()
 {
-    if (!m_settings.accountPtr() && accountId() == 0) {
-        cancelTask(i18nc("@info:status", "Resource is not configured"));
-        Q_EMIT status(NotConfigured, i18nc("@info:status", "Resource is not configured"));
-        return false;
-    }
-
-    return true;
+    return false;
 }
 
 void GoogleResource::slotAuthJobFinished(KGAPI2::Job *job)
diff --git a/resources/google-groupware/googlesettings.cpp b/resources/google-groupware/googlesettings.cpp
index cfb1ccf..038644a 100644
--- a/resources/google-groupware/googlesettings.cpp
+++ b/resources/google-groupware/googlesettings.cpp
@@ -158,17 +158,17 @@ void GoogleSettings::addTaskList(const QString &taskList)
 
 QString GoogleSettings::clientId() const
 {
-    return QStringLiteral("554041944266.apps.googleusercontent.com");
+    return {};
 }
 
 QString GoogleSettings::clientSecret() const
 {
-    return QStringLiteral("mdT1DjzohxN3npUUzkENT0gO");
+    return {};
 }
 
 bool GoogleSettings::isReady() const
 {
-    return m_isReady;
+    return false;
 }
 
 AccountPtr GoogleSettings::accountPtr()
diff --git a/resources/imap/gmailpasswordrequester.cpp b/resources/imap/gmailpasswordrequester.cpp
index bcb433f..1d2b64a 100644
--- a/resources/imap/gmailpasswordrequester.cpp
+++ b/resources/imap/gmailpasswordrequester.cpp
@@ -12,8 +12,8 @@
 #include <KGAPI/AccountManager>
 #include <KGAPI/AuthJob>
 
-#define GOOGLE_API_KEY QStringLiteral("554041944266.apps.googleusercontent.com")
-#define GOOGLE_API_SECRET QStringLiteral("mdT1DjzohxN3npUUzkENT0gO")
+#define GOOGLE_API_KEY QStringLiteral("")
+#define GOOGLE_API_SECRET QStringLiteral("")
 
 GmailPasswordRequester::GmailPasswordRequester(ImapResourceBase *resource, QObject *parent)
     : XOAuthPasswordRequester(parent)
