diff --git a/resources/google-groupware/calendarhandler.cpp b/resources/google-groupware/calendarhandler.cpp
index f306829..70eede4 100644
--- a/resources/google-groupware/calendarhandler.cpp
+++ b/resources/google-groupware/calendarhandler.cpp
@@ -360,7 +360,7 @@ void FreeBusyHandler::retrieveFreeBusy(const QString &email, const QDateTime &st
         }
 
         KCalendarCore::FreeBusy::Ptr fb(new KCalendarCore::FreeBusy);
-        fb->setUid(QStringLiteral("%1%2@google.com").arg(QDateTime::currentDateTimeUtc().toString(QStringLiteral("yyyyMMddTHHmmssZ"))));
+        fb->setUid(QStringLiteral("%1%2@nothanks.invalid").arg(QDateTime::currentDateTimeUtc().toString(QStringLiteral("yyyyMMddTHHmmssZ"))));
         fb->setOrganizer(job->account()->accountName());
         fb->addAttendee(KCalendarCore::Attendee(QString(), queryJob->id()));
         // FIXME: is it really sort?
diff --git a/resources/google-groupware/googleresource.cpp b/resources/google-groupware/googleresource.cpp
index d9e9d95..5383620 100644
--- a/resources/google-groupware/googleresource.cpp
+++ b/resources/google-groupware/googleresource.cpp
@@ -147,7 +147,6 @@ void GoogleResource::configure(WId windowId)
         Q_EMIT configurationDialogAccepted();
 
         if (m_settings->accountPtr().isNull()) {
-            Q_EMIT status(NotConfigured, i18n("Configured account does not exist"));
             m_isConfiguring = false;
             return;
         }
@@ -177,18 +176,10 @@ QList<QUrl> GoogleResource::scopes() const
 
 void GoogleResource::updateResourceName()
 {
-    const QString accountName = m_settings->account();
-    setName(i18nc("%1 is account name (user@gmail.com)", "Google Groupware (%1)", accountName.isEmpty() ? i18n("not configured") : accountName));
 }
 
 void GoogleResource::reloadConfig()
 {
-    const AccountPtr account = m_settings->accountPtr();
-    if (account.isNull() || account->accountName().isEmpty()) {
-        Q_EMIT status(NotConfigured, i18n("Configured account does not exist"));
-    } else {
-        emitReadyStatus();
-    }
 }
 
 bool GoogleResource::handleError(KGAPI2::Job *job, bool _cancelTask)
@@ -257,13 +248,7 @@ void GoogleResource::requestAuthenticationFromUser(const KGAPI2::AccountPtr &acc
 
 bool GoogleResource::canPerformTask()
 {
-    if (!m_settings->accountPtr() && accountId() == 0) {
-        cancelTask(i18nc("@info:status", "Resource is not configured"));
-        Q_EMIT status(NotConfigured, i18nc("@info:status", "Resource is not configured"));
-        return false;
-    }
-
-    return true;
+    return false;
 }
 
 void GoogleResource::slotAuthJobFinished(KGAPI2::Job *job)
diff --git a/resources/google-groupware/googlesettings.cpp b/resources/google-groupware/googlesettings.cpp
index 5da69cf..33404be 100644
--- a/resources/google-groupware/googlesettings.cpp
+++ b/resources/google-groupware/googlesettings.cpp
@@ -138,17 +138,17 @@ void GoogleSettings::addTaskList(const QString &taskList)
 
 QString GoogleSettings::clientId() const
 {
-    return QStringLiteral("554041944266.apps.googleusercontent.com");
+    return {};
 }
 
 QString GoogleSettings::clientSecret() const
 {
-    return QStringLiteral("mdT1DjzohxN3npUUzkENT0gO");
+    return {};
 }
 
 bool GoogleSettings::isReady() const
 {
-    return m_isReady;
+    return false;
 }
 
 AccountPtr GoogleSettings::accountPtr()
diff --git a/resources/imap/gmailpasswordrequester.cpp b/resources/imap/gmailpasswordrequester.cpp
index 496ab4b..0eb0ecf 100644
--- a/resources/imap/gmailpasswordrequester.cpp
+++ b/resources/imap/gmailpasswordrequester.cpp
@@ -12,8 +12,8 @@
 #include <KGAPI/AccountManager>
 #include <KGAPI/AuthJob>
 
-#define GOOGLE_API_KEY QStringLiteral("554041944266.apps.googleusercontent.com")
-#define GOOGLE_API_SECRET QStringLiteral("mdT1DjzohxN3npUUzkENT0gO")
+#define GOOGLE_API_KEY QStringLiteral("")
+#define GOOGLE_API_SECRET QStringLiteral("")
 
 GmailPasswordRequester::GmailPasswordRequester(ImapResourceBase *resource, QObject *parent)
     : PasswordRequesterInterface(parent)
