diff --git a/lib/bundler/lockfile_generator.rb b/lib/bundler/lockfile_generator.rb
index 3bc6bd7..6c8942a 100644
--- a/lib/bundler/lockfile_generator.rb
+++ b/lib/bundler/lockfile_generator.rb
@@ -20,7 +20,6 @@ module Bundler
       add_platforms
       add_dependencies
       add_locked_ruby_version
-      add_bundled_with
 
       out
     end
@@ -71,7 +70,6 @@ module Bundler
     end
 
     def add_bundled_with
-      add_section("BUNDLED WITH", definition.locked_bundler_version.to_s)
     end
 
     def add_section(name, value)
diff --git a/lib/bundler/lockfile_parser.rb b/lib/bundler/lockfile_parser.rb
index f87faff..27d5fa4 100644
--- a/lib/bundler/lockfile_parser.rb
+++ b/lib/bundler/lockfile_parser.rb
@@ -79,8 +79,6 @@ module Bundler
           @state = :platform
         elsif line == RUBY
           @state = :ruby
-        elsif line == BUNDLED
-          @state = :bundled_with
         elsif line =~ /^[^\s]/
           @state = nil
         elsif @state
@@ -227,9 +225,7 @@ module Bundler
     end
 
     def parse_bundled_with(line)
-      line = line.strip
-      return unless Gem::Version.correct?(line)
-      @bundler_version = Gem::Version.create(line)
+      @bundler_version = Gem::Version.create(Bundler::VERSION)
     end
 
     def parse_ruby(line)
