diff --git a/src/mainwindow.cpp b/src/mainwindow.cpp
index dc7856e..8e75fe1 100644
--- a/src/mainwindow.cpp
+++ b/src/mainwindow.cpp
@@ -364,7 +364,7 @@ void MainWindow::makeJoystickTabs()
         ui->tabWidget->addTab(tabwidget, joytabName);
     }
 
-    if (joysticks > 0)
+    if (joysticks != nullptr)
     {
         ui->tabWidget->setCurrentIndex(0);
         ui->stackedWidget->setCurrentIndex(1);
