diff --git a/src/pipewire/conf.c b/src/pipewire/conf.c
index 387a3ab..93a7c5e 100644
--- a/src/pipewire/conf.c
+++ b/src/pipewire/conf.c
@@ -1201,7 +1201,6 @@ int pw_conf_load_conf_for_context(struct pw_properties *props, struct pw_propert
 	if ((res = try_load_conf(conf_prefix, conf_name, conf)) < 0) {
 		conf_name = pw_properties_get(props, PW_KEY_CONFIG_NAME);
 		if (spa_streq(conf_name, "client-rt.conf")) {
-			pw_log_warn("setting config.name to client-rt.conf is deprecated, using client.conf");
 			conf_name = NULL;
 		}
 		if (conf_name == NULL)
