diff --git a/demux/demux_mkv.c b/demux/demux_mkv.c
index 2e03fac874..72a4c64d02 100644
--- a/demux/demux_mkv.c
+++ b/demux/demux_mkv.c
@@ -740,6 +740,14 @@ static void parse_trackentry(struct demuxer *demuxer,
         MP_DBG(demuxer, "|  + Forced flag: %d\n", track->forced_track);
     }
 
+    // unset default track flags on subtitles
+    // (fixes the issue where the given language is not in the preferred language list [config file]
+    //  and mpv than just selects the subtitle marked as "default" as fallback)
+    if (entry->track_type == MATROSKA_TRACK_SUBTITLE) {
+        track->default_track = 0;
+        track->forced_track = 0;
+    }
+
     if (entry->n_default_duration) {
         track->default_duration = entry->default_duration / 1e9;
         if (entry->default_duration == 0) {
