diff --git a/src/context.cpp b/src/context.cpp
index 51bf7e0..054f0eb 100644
--- a/src/context.cpp
+++ b/src/context.cpp
@@ -95,9 +95,6 @@ static void source_cb(pa_context *context, const pa_source_info *info, int eol,
 {
     if (!isGoodState(eol))
         return;
-    // FIXME: This forces excluding monitors
-    if (info->monitor_of_sink != PA_INVALID_INDEX)
-        return;
     Q_ASSERT(context);
     Q_ASSERT(data);
     ((Context *)data)->sourceCallback(info);
