diff --git a/lib/rubygems.rb b/lib/rubygems.rb
index 3f4106f..73560f9 100644
--- a/lib/rubygems.rb
+++ b/lib/rubygems.rb
@@ -1324,6 +1324,7 @@ def default_gem_load_paths
   # Defaults the operating system (or packager) wants to provide for RubyGems.
 
   require "rubygems/defaults/operating_system"
+  require "rubygems/defaults/operating_system_user"
 rescue LoadError
   # Ignored
 rescue StandardError => e
diff --git a/lib/rubygems/defaults/operating_system_user.rb b/lib/rubygems/defaults/operating_system_user.rb
new file mode 100644
index 0000000..75f7948
--- /dev/null
+++ b/lib/rubygems/defaults/operating_system_user.rb
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Gem
+  class << self
+    def gentoo_bindir
+      Process.euid.zero? ? '/usr/local/bin' : File.join(user_home, ".gem/bin/#{RbConfig::CONFIG["ruby_version"]}")
+    end
+
+    undef :operating_system_defaults
+    def operating_system_defaults
+      options = "--install-dir #{gentoo_install_dir} --bindir #{gentoo_bindir}"
+
+      {
+        'install' => options,
+        'uninstall' => options,
+        'update' => options
+      }
+    end
+  end
+end
