diff --git a/gui/trayitem.cpp b/gui/trayitem.cpp
index 8514ba0..f57437a 100644
--- a/gui/trayitem.cpp
+++ b/gui/trayitem.cpp
@@ -145,7 +145,8 @@ void TrayItem::setup()
 	QIcon icon;
 	icon.addFile(CANTATA_SYS_ICONS_DIR + PROJECT_REV_ID ".png");
 #else
-	QIcon icon = QIcon::fromTheme(Utils::Gnome == Utils::currentDe() ? PROJECT_REV_ID "-symbolic" : PROJECT_REV_ID);
+	QIcon icon;
+	icon.addFile("/usr/share/icons/hicolor/scalable/apps/dog.unix.cantata.Cantata.svg");
 	// Bug: 660 If installed to non-standard folder, QIcon::fromTheme does not seem to find icon. Therefore
 	// add icon files here...
 	if (icon.isNull()) {
diff --git a/support/.QtAwesome.cpp.kate-swp b/support/.QtAwesome.cpp.kate-swp
deleted file mode 100644
index ae3cfc1..0000000
Binary files a/support/.QtAwesome.cpp.kate-swp and /dev/null differ
diff --git a/support/QtAwesome.cpp b/support/QtAwesome.cpp
index 680b983..9ddb782 100644
--- a/support/QtAwesome.cpp
+++ b/support/QtAwesome.cpp
@@ -472,6 +472,7 @@ QFont QtAwesome::font(int style, int size) const
 	QFont font(_fontDetails[style].fontFamily());
 	font.setPixelSize(size);
 	font.setWeight(_fontDetails[style].fontWeight());
+	font.setStyleName("Solid");
 
 	return font;
 }
