diff --git a/configure.ac b/configure.ac
index d81e74a..5cd96d0 100644
--- a/configure.ac
+++ b/configure.ac
@@ -124,10 +124,10 @@ dnl -------------------------------------------------------------------------
 dnl Common
 dnl
 if test "$enable_shared" = "yes"; then
-    ZenLib_LIBS="-L$(pwd)/.libs -lzen -lpthread -lstdc++ $ZenLib_LIBS"
+    ZenLib_LIBS="-L$(pwd)/.libs -lzen -lpthread -lc++ $ZenLib_LIBS"
 fi
 if test "$enable_static" = "yes"; then
-    ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a -lpthread -lstdc++ $ZenLib_LIBS_Static"
+    ZenLib_LIBS_Static="$(pwd)/.libs/libzen.a -lpthread -lc++ $ZenLib_LIBS_Static"
 fi
 
 dnl #########################################################################
@@ -232,7 +232,7 @@ fi
 dnl -------------------------------------------------------------------------
 dnl External libs
 dnl
-LDFLAGS="$LDFLAGS -lpthread -lstdc++"
+LDFLAGS="$LDFLAGS -lpthread -lc++"
 
 dnl #########################################################################
 dnl ### Output
diff --git a/libzen.pc.in b/libzen.pc.in
index c3e6778..eccc2ef 100644
--- a/libzen.pc.in
+++ b/libzen.pc.in
@@ -9,5 +9,5 @@ Libs_Static=@libdir@/libzen.a -lpthread
 Name: libzen
 Version: @PACKAGE_VERSION@
 Description: ZenLib
-Libs: -L${libdir} -lzen -lpthread -lstdc++
+Libs: -L${libdir} -lzen -lpthread -lc++
 Cflags: -I${includedir} @ZenLib_CXXFLAGS@
