diff --git a/src/modules/motion_est/Makefile b/src/modules/motion_est/Makefile
index 0d14275..acd12ce 100644
--- a/src/modules/motion_est/Makefile
+++ b/src/modules/motion_est/Makefile
@@ -1,4 +1,4 @@
-CFLAGS += -I../.. 
+CFLAGS += -I../.. -std=gnu11
 
 LDFLAGS += -L../../framework -lmlt -lm
 
diff --git a/src/modules/opengl/Makefile b/src/modules/opengl/Makefile
index 65a7b34..46a0be9 100644
--- a/src/modules/opengl/Makefile
+++ b/src/modules/opengl/Makefile
@@ -1,6 +1,7 @@
 include ../../../config.mak
 
 CFLAGS := -I../.. $(CFLAGS)
+CXXFLAGS := -std=c++11
 
 LDFLAGS := -L../../framework -lmlt -lm $(LDFLAGS)
 
