diff --git a/apps/lib/katemainwindow.cpp b/apps/lib/katemainwindow.cpp
index af3414c..7b6c8af 100644
--- a/apps/lib/katemainwindow.cpp
+++ b/apps/lib/katemainwindow.cpp
@@ -1055,18 +1055,6 @@ void KateMainWindow::slotDropEvent(QDropEvent *event)
     if (event->mimeData()->hasUrls()) {
         QList<QUrl> textlist = event->mimeData()->urls();
 
-        // Try to get the KTextEditor::View that sent this, and activate it, so that the file opens in the
-        // view where it was dropped
-        auto *kVsender = qobject_cast<KTextEditor::View *>(QObject::sender());
-        if (kVsender != nullptr) {
-            if (auto parent = kVsender->parent()) {
-                auto *vs = qobject_cast<KateViewSpace *>(parent->parent());
-                if (vs != nullptr) {
-                    m_viewManager->setActiveSpace(vs);
-                }
-            }
-        }
-
         for (const QUrl &url : std::as_const(textlist)) {
             // if url has no file component, try and recursively scan dir
             KFileItem kitem(url);
@@ -1084,7 +1072,7 @@ void KateMainWindow::slotDropEvent(QDropEvent *event)
                     connect(list_job, &KIO::ListJob::entries, this, &KateMainWindow::slotListRecursiveEntries);
                 }
             } else {
-                m_viewManager->openUrl(url);
+                m_viewManager->openUrlInNewWindow(url);
             }
         }
     }
diff --git a/apps/lib/kateviewmanager.cpp b/apps/lib/kateviewmanager.cpp
index f3f3b52..432fbc5 100644
--- a/apps/lib/kateviewmanager.cpp
+++ b/apps/lib/kateviewmanager.cpp
@@ -33,6 +33,7 @@
 #include <QFileDialog>
 #include <QScrollBar>
 #include <QTimer>
+#include <QProcess>
 
 // END Includes
 
@@ -549,6 +550,12 @@ void KateViewManager::openUrl(const QUrl &url)
     openUrl(url, QString());
 }
 
+void KateViewManager::openUrlInNewWindow(const QUrl &url)
+{
+    // HACK: just force a new instance to make absolutely sure a new window is opened
+    QProcess::startDetached(QStringLiteral("/usr/bin/kwrite"), {url.toString()}, QDir::currentPath());
+}
+
 void KateViewManager::openUrlOrProject(const QUrl &url)
 {
     if (!url.isLocalFile()) {
diff --git a/apps/lib/kateviewmanager.h b/apps/lib/kateviewmanager.h
index c2cca52..130daaa 100644
--- a/apps/lib/kateviewmanager.h
+++ b/apps/lib/kateviewmanager.h
@@ -74,6 +74,7 @@ public:
 
 public:
     void openUrl(const QUrl &url);
+    void openUrlInNewWindow(const QUrl &url);
     void openUrlOrProject(const QUrl &url);
     void openRecent(const QUrl &url);
     void addPositionToHistory(const QUrl &url, KTextEditor::Cursor pos);
