diff --git a/svcore/data/model/NoteModel.h b/svcore/data/model/NoteModel.h
index 3ae2485..e648903 100644
--- a/svcore/data/model/NoteModel.h
+++ b/svcore/data/model/NoteModel.h
@@ -403,8 +403,8 @@ public:
              .arg(m_events.getExportId())
              .arg(m_subtype == FLEXI_NOTE ? "flexinote" : "note")
              .arg(m_valueQuantization)
-             .arg(m_valueMinimum)
-             .arg(m_valueMaximum)
+             .arg(m_valueMinimum.load())
+             .arg(m_valueMaximum.load())
              .arg(encodeEntities(m_units))
              .arg(extraAttributes));
         
diff --git a/svcore/data/model/RegionModel.h b/svcore/data/model/RegionModel.h
index 916a047..db021ae 100644
--- a/svcore/data/model/RegionModel.h
+++ b/svcore/data/model/RegionModel.h
@@ -335,8 +335,8 @@ public:
              .arg(m_events.getExportId())
              .arg("region")
              .arg(m_valueQuantization)
-             .arg(m_valueMinimum)
-             .arg(m_valueMaximum)
+             .arg(m_valueMinimum.load())
+             .arg(m_valueMaximum.load())
              .arg(encodeEntities(m_units))
              .arg(extraAttributes));
         
diff --git a/svcore/data/model/SparseTimeValueModel.h b/svcore/data/model/SparseTimeValueModel.h
index fe6e70c..7033503 100644
--- a/svcore/data/model/SparseTimeValueModel.h
+++ b/svcore/data/model/SparseTimeValueModel.h
@@ -342,8 +342,8 @@ public:
              .arg("true") // always true after model reaches 100% -
                           // subsequent events are always notified
              .arg(m_events.getExportId())
-             .arg(m_valueMinimum)
-             .arg(m_valueMaximum)
+             .arg(m_valueMinimum.load())
+             .arg(m_valueMaximum.load())
              .arg(encodeEntities(m_units))
              .arg(extraAttributes));
         
