diff --git a/bqfft/build/Makefile.inc b/bqfft/build/Makefile.inc
index b45a49f..be73bf8 100644
--- a/bqfft/build/Makefile.inc
+++ b/bqfft/build/Makefile.inc
@@ -14,7 +14,7 @@ TEST_OBJECTS	:= $(TEST_SOURCES:.cpp=.o)
 
 OPTFLAGS ?= -O3 -ffast-math
 
-CXXFLAGS ?= -std=c++98 -Wall -Wextra -Werror $(FFT_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES) -I../bqvec -fpic
+CXXFLAGS ?= -std=c++98 -Wall -Wextra $(FFT_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES) -I../bqvec -fpic
 
 LIBRARY	:= libbqfft.a
 
diff --git a/bqresample/build/Makefile.inc b/bqresample/build/Makefile.inc
index a314994..02029e2 100644
--- a/bqresample/build/Makefile.inc
+++ b/bqresample/build/Makefile.inc
@@ -15,7 +15,7 @@ TEST_OBJECTS	:= $(TEST_SOURCES:.cpp=.o)
 
 OPTFLAGS ?= -O3 -ffast-math
 
-CXXFLAGS ?= -std=c++98 -fpic -Wall -Wextra -Werror $(RESAMPLE_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES) -I../bqvec
+CXXFLAGS ?= -std=c++98 -fpic -Wall -Wextra $(RESAMPLE_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES) -I../bqvec
 
 CFLAGS	:= -fpic -Wall -Wextra $(RESAMPLE_DEFINES) $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) $(OPTFLAGS) -I. $(THIRD_PARTY_INCLUDES)
 
diff --git a/bqvec/build/Makefile.inc b/bqvec/build/Makefile.inc
index 67fcb4b..5c1ec33 100644
--- a/bqvec/build/Makefile.inc
+++ b/bqvec/build/Makefile.inc
@@ -17,7 +17,7 @@ TEST_OBJECTS	:= $(TEST_SOURCES:.cpp=.o)
 
 OPTFLAGS := -O3 -ffast-math
 
-CXXFLAGS := -std=c++98 -fpic -Wall -Wextra -Werror $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) -I. $(THIRD_PARTY_INCLUDES) -I$(HEADER_DIR) $(OPTFLAGS)
+CXXFLAGS := -std=c++98 -fpic -Wall -Wextra $(VECTOR_DEFINES) $(ALLOCATOR_DEFINES) -I. $(THIRD_PARTY_INCLUDES) -I$(HEADER_DIR) $(OPTFLAGS)
 
 LIBRARY	:= libbqvec.a
 
diff --git a/checker/checker.pri b/checker/checker.pri
index 6f857e1..d95d21a 100644
--- a/checker/checker.pri
+++ b/checker/checker.pri
@@ -2,9 +2,6 @@
 CONFIG += qt stl c++11 exceptions console warn_on
 QT -= xml network gui widgets
 
-!win32 {
-    QMAKE_CXXFLAGS += -Werror
-}
 
 OBJECTS_DIR = o
 MOC_DIR = o
diff --git a/checker/helper.pro b/checker/helper.pro
index e23fe3d..67227d2 100644
--- a/checker/helper.pro
+++ b/checker/helper.pro
@@ -10,9 +10,6 @@ win32-x-g++:QMAKE_LFLAGS += -Wl,-subsystem,console
 
 macx*: CONFIG -= app_bundle
 
-!win32* {
-    QMAKE_CXXFLAGS += -Werror
-}
 
 linux* {
     LIBS += -ldl
diff --git a/configure.ac b/configure.ac
index d3240da..cf605d1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -54,7 +54,7 @@ CXXFLAGS_MINIMAL="$AUTOCONF_CXXFLAGS"
 
 if test "x$GCC" = "xyes"; then
    	CXXFLAGS_ANY="-std=c++11 -fpic -Wall -Wextra -Woverloaded-virtual -Wformat-nonliteral -Wformat-security -Winit-self -Wswitch-enum -Wconversion -pipe"
-        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g -O2 -Werror"
+        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g -O2"
    	CXXFLAGS_RELEASE="$CXXFLAGS_ANY -g0 -O3 -ffast-math"
    	CXXFLAGS_MINIMAL="$CXXFLAGS_ANY -g0 -O0"
         LIBS_DEBUG="$LIBS -lubsan"
diff --git a/convert.pro b/convert.pro
index 08a0852..9766b9d 100644
--- a/convert.pro
+++ b/convert.pro
@@ -15,8 +15,6 @@ exists(config.pri) {
     macx*: LIBS -= -framework CoreAudio -framework CoreMidi -framework AudioUnit -framework AudioToolbox -framework CoreFoundation -framework CoreServices -framework Accelerate -lbz2 -lz
 }
 
-# Can't support this flag with the JSON11 and basen modules as they stand
-QMAKE_CXXFLAGS -= -Werror
 
 # Using the "console" CONFIG flag above should ensure this happens for
 # normal Windows builds, but this may be necessary when cross-compiling
diff --git a/piper-cpp/Makefile b/piper-cpp/Makefile
index 3e75671..dc4cfc1 100644
--- a/piper-cpp/Makefile
+++ b/piper-cpp/Makefile
@@ -7,7 +7,7 @@ INCFLAGS	:= -Iext -Iext/sord -Iext/serd -I$(VAMPSDK_DIR) -I. -I/usr/local/includ
 OPTFLAGS	:= -O3 -fPIC
 
 CFLAGS		:= -Wall -std=c99 $(OPTFLAGS) $(INCFLAGS)
-CXXFLAGS	:= -Wall -Wextra -Werror -Wno-error=unused-parameter -std=c++11 $(OPTFLAGS) $(INCFLAGS)
+CXXFLAGS	:= -Wall -Wextra -Wno-error=unused-parameter -std=c++11 $(OPTFLAGS) $(INCFLAGS)
 
 LDFLAGS		:= $(VAMPSDK_DIR)/libvamp-hostsdk.a -L/usr/local/lib -lcapnp -lkj 
 
diff --git a/piper-cpp/ext/json11/CMakeLists.txt b/piper-cpp/ext/json11/CMakeLists.txt
index 819c36f..6729cf3 100644
--- a/piper-cpp/ext/json11/CMakeLists.txt
+++ b/piper-cpp/ext/json11/CMakeLists.txt
@@ -29,7 +29,7 @@ target_compile_options(json11
 
 # Set warning flags, which may vary per platform
 include(CheckCXXCompilerFlag)
-set(_possible_warnings_flags /W4 /WX -Wextra -Werror)
+set(_possible_warnings_flags /W4 /WX -Wextra)
 foreach(_warning_flag in ${_possible_warnings_flags})
   CHECK_CXX_COMPILER_FLAG(_warning_flag _flag_supported)
   if(${_flag_supported})
diff --git a/piper-cpp/vamp-client/qt/test.pro b/piper-cpp/vamp-client/qt/test.pro
index b4ed963..6b7c1b5 100644
--- a/piper-cpp/vamp-client/qt/test.pro
+++ b/piper-cpp/vamp-client/qt/test.pro
@@ -4,9 +4,6 @@ TEMPLATE = app
 CONFIG += qt stl c++11 exceptions console warn_on
 QT -= xml network gui widgets
 
-!win32 {
-    QMAKE_CXXFLAGS += -Werror
-}
 
 OBJECTS_DIR = ../o
 MOC_DIR = ../o
diff --git a/platform-dataquay.pri b/platform-dataquay.pri
index b6ebd78..56c4903 100644
--- a/platform-dataquay.pri
+++ b/platform-dataquay.pri
@@ -7,7 +7,6 @@ CONFIG += staticlib c++11
 
 DEFINES -= USE_REDLAND
 QMAKE_CXXFLAGS -= -I/usr/include/rasqal -I/usr/include/raptor2
-QMAKE_CXXFLAGS -= -Werror
 EXTRALIBS -= -lrdf
 
 DEFINES += USE_SORD
diff --git a/server.pro b/server.pro
index fd55fce..f60a3ea 100644
--- a/server.pro
+++ b/server.pro
@@ -15,8 +15,6 @@ exists(config.pri) {
     macx*: LIBS -= -framework CoreAudio -framework CoreMidi -framework AudioUnit -framework AudioToolbox -framework CoreFoundation -framework CoreServices -framework Accelerate -lbz2 -lz
 }
 
-# Can't support this flag with the JSON11 and basen modules as they stand
-QMAKE_CXXFLAGS -= -Werror
 
 # Using the "console" CONFIG flag above should ensure this happens for
 # normal Windows builds, but this may be necessary when cross-compiling
diff --git a/svapp/configure.ac b/svapp/configure.ac
index fc7b875..2472023 100644
--- a/svapp/configure.ac
+++ b/svapp/configure.ac
@@ -54,7 +54,7 @@ CXXFLAGS_MINIMAL="$AUTOCONF_CXXFLAGS"
 
 if test "x$GCC" = "xyes"; then
    	CXXFLAGS_ANY="-Wall -Wextra -Woverloaded-virtual -Wformat-nonliteral -Wformat-security -Winit-self -Wswitch-enum -Wconversion -pipe"
-        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -Werror -g"
+        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g"
    	CXXFLAGS_RELEASE="$CXXFLAGS_ANY -g0 -O2"
    	CXXFLAGS_MINIMAL="$CXXFLAGS_ANY -g0 -O0"
 fi
diff --git a/svcore/configure.ac b/svcore/configure.ac
index 42c882f..172d73b 100644
--- a/svcore/configure.ac
+++ b/svcore/configure.ac
@@ -54,7 +54,7 @@ CXXFLAGS_MINIMAL="$AUTOCONF_CXXFLAGS"
 
 if test "x$GCC" = "xyes"; then
    	CXXFLAGS_ANY="-Wall -Wextra -Woverloaded-virtual -Wformat-nonliteral -Wformat-security -Winit-self -Wswitch-enum -Wconversion -pipe"
-        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -Werror -g"
+        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g"
    	CXXFLAGS_RELEASE="$CXXFLAGS_ANY -g0 -O2"
    	CXXFLAGS_MINIMAL="$CXXFLAGS_ANY -g0 -O0"
 fi
diff --git a/svgui/configure.ac b/svgui/configure.ac
index 8104296..9891961 100644
--- a/svgui/configure.ac
+++ b/svgui/configure.ac
@@ -54,7 +54,7 @@ CXXFLAGS_MINIMAL="$AUTOCONF_CXXFLAGS"
 
 if test "x$GCC" = "xyes"; then
    	CXXFLAGS_ANY="-Wall -Wextra -Woverloaded-virtual -Wformat-nonliteral -Wformat-security -Winit-self -Wswitch-enum -Wconversion -pipe"
-        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -Werror -g"
+        CXXFLAGS_DEBUG="$CXXFLAGS_ANY -g"
    	CXXFLAGS_RELEASE="$CXXFLAGS_ANY -g0 -O2"
    	CXXFLAGS_MINIMAL="$CXXFLAGS_ANY -g0 -O0"
 fi
