diff --git a/iptables/ip6tables.c b/iptables/ip6tables.c
index c95355b..8c686e8 100644
--- a/iptables/ip6tables.c
+++ b/iptables/ip6tables.c
@@ -1067,7 +1067,7 @@ int do_command6(int argc, char *argv[], char **table,
 	struct in6_addr *smasks = NULL, *dmasks = NULL;
 
 	int verbose = 0;
-	int wait = 0;
+	int wait = 10;
 	struct timeval wait_interval = {
 		.tv_sec	= 1,
 	};
diff --git a/iptables/iptables.c b/iptables/iptables.c
index 7d61831..5e8867b 100644
--- a/iptables/iptables.c
+++ b/iptables/iptables.c
@@ -1066,7 +1066,7 @@ int do_command4(int argc, char *argv[], char **table,
 	};
 	bool wait_interval_set = false;
 	int verbose = 0;
-	int wait = 0;
+	int wait = 10;
 	const char *chain = NULL;
 	const char *shostnetworkmask = NULL, *dhostnetworkmask = NULL;
 	const char *policy = NULL, *newname = NULL;
diff --git a/iptables/xshared.c b/iptables/xshared.c
index 71f6899..a5184a9 100644
--- a/iptables/xshared.c
+++ b/iptables/xshared.c
@@ -323,7 +323,7 @@ int xtables_lock_or_exit(int wait, struct timeval *wait_interval)
 
 int parse_wait_time(int argc, char *argv[])
 {
-	int wait = -1;
+	int wait = 10;
 
 	if (optarg) {
 		if (sscanf(optarg, "%i", &wait) != 1)
