diff --git a/deps/cares/configure b/deps/cares/configure
index 2f182e0c..a6a590c3 100755
--- a/deps/cares/configure
+++ b/deps/cares/configure
@@ -24844,7 +24844,7 @@ if test ${enable_libgcc+y}
 then :
   enableval=$enable_libgcc;  case "$enableval" in
   yes)
-        LIBS="$LIBS -lgcc"
+        LIBS="$LIBS"
        { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 printf "%s\n" "yes" >&6; }
        ;;
diff --git a/deps/cares/configure.ac b/deps/cares/configure.ac
index 54e79d6e..aba71ea5 100644
--- a/deps/cares/configure.ac
+++ b/deps/cares/configure.ac
@@ -473,7 +473,7 @@ AC_ARG_ENABLE(libgcc,
 AS_HELP_STRING([--enable-libgcc],[use libgcc when linking]),
 [ case "$enableval" in
   yes)
-        LIBS="$LIBS -lgcc"
+        LIBS="$LIBS"
        AC_MSG_RESULT(yes)
        ;;
   *)   AC_MSG_RESULT(no)
