diff --git a/kconf_update/gtktheme.cpp b/kconf_update/gtktheme.cpp
index a18f2f6..c55057b 100644
--- a/kconf_update/gtktheme.cpp
+++ b/kconf_update/gtktheme.cpp
@@ -11,28 +11,14 @@
 
 #include "configeditor.h"
 
-QString gtk2Theme();
-void upgradeGtk2Theme();
 void upgradeGtk3Theme();
 
 int main()
 {
-    upgradeGtk2Theme();
     upgradeGtk3Theme();
     return 0;
 }
 
-void upgradeGtk2Theme()
-{
-    QString currentGtk2Theme = gtk2Theme();
-    if (currentGtk2Theme.isEmpty() //
-        || currentGtk2Theme == QStringLiteral("oxygen-gtk") //
-        || currentGtk2Theme == QStringLiteral("BreezyGTK") //
-        || currentGtk2Theme == QStringLiteral("Orion")) {
-        ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-theme-name"), QStringLiteral("Breeze"));
-    }
-}
-
 void upgradeGtk3Theme()
 {
     QString currentGtk3Theme = ConfigEditor::gtk3ConfigValueSettingsIni(QStringLiteral("gtk-theme-name"));
@@ -45,20 +31,3 @@ void upgradeGtk3Theme()
         ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Net/ThemeName"), QStringLiteral("Breeze"));
     }
 }
-
-QString gtk2Theme()
-{
-    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
-    QFile gtkrc(gtkrcPath);
-    if (gtkrc.open(QIODevice::ReadWrite | QIODevice::Text)) {
-        const QRegularExpression regExp(QStringLiteral("gtk-theme-name=[^\n]*($|\n)"));
-        while (!gtkrc.atEnd()) {
-            QString line = gtkrc.readLine();
-            if (line.contains(regExp)) {
-                return line.split('"')[1];
-            }
-        }
-    }
-
-    return QString();
-}
diff --git a/kded/configeditor.cpp b/kded/configeditor.cpp
index c3476b7..fb2120d 100644
--- a/kded/configeditor.cpp
+++ b/kded/configeditor.cpp
@@ -103,20 +103,6 @@ void ConfigEditor::setGtk3ConfigValueXSettingsd(const QString &paramName, const
     reloadXSettingsd();
 }
 
-void ConfigEditor::setGtk2ConfigValue(const QString &paramName, const QVariant &paramValue)
-{
-    QString gtkrcPath = qEnvironmentVariable("GTK2_RC_FILES", QDir::homePath() + QStringLiteral("/.gtkrc-2.0"));
-    if (gtkrcPath.contains(QStringLiteral(":/"))) { // I.e. env variable contains multiple paths
-        gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
-    }
-    QFile gtkrc(gtkrcPath);
-    QString gtkrcContents = readFileContents(gtkrc);
-    replaceValueInGtkrcContents(gtkrcContents, paramName, paramValue);
-    gtkrc.remove();
-    gtkrc.open(QIODevice::WriteOnly | QIODevice::Text);
-    gtkrc.write(gtkrcContents.toUtf8());
-}
-
 void ConfigEditor::setCustomClientSideDecorations(const QStringList &windowDecorationsButtonsImages)
 {
     saveWindowDecorationsToAssets(windowDecorationsButtonsImages);
@@ -136,33 +122,6 @@ void ConfigEditor::setGtk3Colors(const QMap<QString, QColor> &colorsDefinitions)
     addGtkModule(QStringLiteral("colorreload-gtk-module"));
 }
 
-void ConfigEditor::removeLegacyGtk2Strings()
-{
-    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
-    QFile gtkrc(gtkrcPath);
-    QString gtkrcContents = readFileContents(gtkrc);
-
-    // Remove "include" lines
-    // Example:
-    // include "/usr/share/themes/Adwaita-dark/gtk-2.0/gtkrc"
-    static const QRegularExpression includeRegExp(QStringLiteral("include .*\n"));
-    gtkrcContents.remove(includeRegExp);
-
-    // Remove redundant font config lines
-    // Example:
-    // style "user-font"
-    // {
-    //     font_name="Noto Sans Regular"
-    // }
-    // widget_class "*" style "user-font"
-    static const QRegularExpression userFontStyleRegexp(QStringLiteral("style(.|\n)*{(.|\n)*}\nwidget_class.*\"user-font\""));
-    gtkrcContents.remove(userFontStyleRegexp);
-
-    gtkrc.remove();
-    gtkrc.open(QIODevice::WriteOnly | QIODevice::Text);
-    gtkrc.write(gtkrcContents.toUtf8());
-}
-
 void ConfigEditor::saveWindowDecorationsToAssets(const QStringList &windowDecorationsButtonsImages)
 {
     QDir assetsFolder{QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation) + QStringLiteral("/gtk-3.0/assets")};
diff --git a/kded/configeditor.h b/kded/configeditor.h
index 72b525a..2bc3716 100644
--- a/kded/configeditor.h
+++ b/kded/configeditor.h
@@ -17,8 +17,6 @@ namespace ConfigEditor
 {
 using namespace std::placeholders;
 
-void setGtk2ConfigValue(const QString &paramName, const QVariant &paramValue);
-
 void setGtkConfigValueGSettings(const QString &paramName, const QVariant &paramValue, const QString &category = QStringLiteral("org.gnome.desktop.interface"));
 void setGtkConfigValueGSettingsAsEnum(const QString &paramName, int paramValue, const QString &category = QStringLiteral("org.gnome.desktop.interface"));
 void setGtk3ConfigValueXSettingsd(const QString &paramName, const QVariant &paramValue);
@@ -38,8 +36,6 @@ void setCustomClientSideDecorations(const QStringList &windowDecorationsButtonsI
 void disableCustomClientSideDecorations();
 void setGtk3Colors(const QMap<QString, QColor> &colorsDefinitions);
 
-void removeLegacyGtk2Strings();
-
 void saveWindowDecorationsToAssets(const QStringList &windowDecorationsButtonsImages);
 void addWindowDecorationsCssFile();
 
diff --git a/kded/gtkconfig.cpp b/kded/gtkconfig.cpp
index 90f629a..bc9e1ba 100644
--- a/kded/gtkconfig.cpp
+++ b/kded/gtkconfig.cpp
@@ -35,7 +35,6 @@ GtkConfig::GtkConfig(QObject *parent, const QVariantList &)
     connect(kcminputConfigWatcher.data(), &KConfigWatcher::configChanged, this, &GtkConfig::onKCMInputSettingsChange);
     connect(breezeConfigWatcher.data(), &KConfigWatcher::configChanged, this, &GtkConfig::onBreezeSettingsChange);
 
-    ConfigEditor::removeLegacyGtk2Strings();
     applyAllSettings();
 }
 
@@ -48,7 +47,6 @@ GtkConfig::~GtkConfig()
 
 void GtkConfig::setGtkTheme(const QString &themeName) const
 {
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-theme-name"), themeName);
     ConfigEditor::setGtkConfigValueGSettings(QStringLiteral("gtk-theme"), themeName);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-theme-name"), themeName);
     ConfigEditor::setGtk4ConfigValueSettingsIni(QStringLiteral("gtk-theme-name"), themeName);
@@ -71,7 +69,6 @@ void GtkConfig::showGtkThemePreview(const QString &themeName) const
 void GtkConfig::setFont() const
 {
     const QString configFontName = configValueProvider->fontName();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-font-name"), configFontName);
     ConfigEditor::setGtkConfigValueGSettings(QStringLiteral("font-name"), configFontName);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-font-name"), configFontName);
     ConfigEditor::setGtk4ConfigValueSettingsIni(QStringLiteral("gtk-font-name"), configFontName);
@@ -81,7 +78,6 @@ void GtkConfig::setFont() const
 void GtkConfig::setIconTheme() const
 {
     const QString iconThemeName = configValueProvider->iconThemeName();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-icon-theme-name"), iconThemeName);
     ConfigEditor::setGtkConfigValueGSettings(QStringLiteral("icon-theme"), iconThemeName);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-icon-theme-name"), iconThemeName);
     ConfigEditor::setGtk4ConfigValueSettingsIni(QStringLiteral("gtk-icon-theme-name"), iconThemeName);
@@ -91,7 +87,6 @@ void GtkConfig::setIconTheme() const
 void GtkConfig::setCursorTheme() const
 {
     const QString cursorThemeName = configValueProvider->cursorThemeName();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-cursor-theme-name"), cursorThemeName);
     ConfigEditor::setGtkConfigValueGSettings(QStringLiteral("cursor-theme"), cursorThemeName);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-cursor-theme-name"), cursorThemeName);
     ConfigEditor::setGtk4ConfigValueSettingsIni(QStringLiteral("gtk-cursor-theme-name"), cursorThemeName);
@@ -101,7 +96,6 @@ void GtkConfig::setCursorTheme() const
 void GtkConfig::setCursorSize() const
 {
     const int cursorSize = configValueProvider->cursorSize();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-cursor-theme-size"), cursorSize);
     ConfigEditor::setGtkConfigValueGSettings(QStringLiteral("cursor-size"), cursorSize);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-cursor-theme-size"), cursorSize);
     ConfigEditor::setGtk4ConfigValueSettingsIni(QStringLiteral("gtk-cursor-theme-size"), cursorSize);
@@ -111,7 +105,6 @@ void GtkConfig::setCursorSize() const
 void GtkConfig::setIconsOnButtons() const
 {
     const bool iconsOnButtonsConfigValue = configValueProvider->iconsOnButtons();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-button-images"), iconsOnButtonsConfigValue);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-button-images"), iconsOnButtonsConfigValue);
     // Deprecated in Gtk4
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/ButtonImages"), iconsOnButtonsConfigValue);
@@ -120,7 +113,6 @@ void GtkConfig::setIconsOnButtons() const
 void GtkConfig::setIconsInMenus() const
 {
     const bool iconsInMenusConfigValue = configValueProvider->iconsInMenus();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-menu-images"), iconsInMenusConfigValue);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-menu-images"), iconsInMenusConfigValue);
     // Deprecated in Gtk4
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/MenuImages"), iconsInMenusConfigValue);
@@ -129,7 +121,6 @@ void GtkConfig::setIconsInMenus() const
 void GtkConfig::setToolbarStyle() const
 {
     const int toolbarStyle = configValueProvider->toolbarStyle();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-toolbar-style"), toolbarStyle);
     ConfigEditor::setGtkConfigValueGSettingsAsEnum(QStringLiteral("toolbar-style"), toolbarStyle);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-toolbar-style"), toolbarStyle);
     // Deprecated in Gtk4
@@ -139,7 +130,6 @@ void GtkConfig::setToolbarStyle() const
 void GtkConfig::setScrollbarBehavior() const
 {
     const bool scrollbarBehavior = configValueProvider->scrollbarBehavior();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-primary-button-warps-slider"), scrollbarBehavior);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-primary-button-warps-slider"), scrollbarBehavior);
     ConfigEditor::setGtk4ConfigValueSettingsIni(QStringLiteral("gtk-primary-button-warps-slider"), scrollbarBehavior);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/PrimaryButtonWarpsSlider"), scrollbarBehavior);
@@ -174,7 +164,6 @@ void GtkConfig::setWindowDecorationsButtonsOrder() const
 void GtkConfig::setEnableAnimations() const
 {
     const bool enableAnimations = configValueProvider->enableAnimations();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-enable-animations"), enableAnimations);
     ConfigEditor::setGtkConfigValueGSettings(QStringLiteral("enable-animations"), enableAnimations);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-enable-animations"), enableAnimations);
     ConfigEditor::setGtk4ConfigValueSettingsIni(QStringLiteral("gtk-enable-animations"), enableAnimations);
