diff --git a/kconf_update/gtktheme.cpp b/kconf_update/gtktheme.cpp
index cf53036..1872049 100644
--- a/kconf_update/gtktheme.cpp
+++ b/kconf_update/gtktheme.cpp
@@ -14,28 +14,14 @@
 #include "config_editor/settings_ini.h"
 #include "config_editor/xsettings.h"
 
-QString gtk2Theme();
-void upgradeGtk2Theme();
 void upgradeGtk3Theme();
 
 int main()
 {
-    upgradeGtk2Theme();
     upgradeGtk3Theme();
     return 0;
 }
 
-void upgradeGtk2Theme()
-{
-    QString currentGtk2Theme = gtk2Theme();
-    if (currentGtk2Theme.isEmpty() //
-        || currentGtk2Theme == QStringLiteral("oxygen-gtk") //
-        || currentGtk2Theme == QStringLiteral("BreezyGTK") //
-        || currentGtk2Theme == QStringLiteral("Orion")) {
-        Gtk2ConfigEditor::setValue(QStringLiteral("gtk-theme-name"), QStringLiteral("Breeze"));
-    }
-}
-
 void upgradeGtk3Theme()
 {
     QString currentGtk3Theme = SettingsIniEditor::value(QStringLiteral("gtk-theme-name"), 3);
@@ -48,20 +34,3 @@ void upgradeGtk3Theme()
         XSettingsEditor::setValue(QStringLiteral("Net/ThemeName"), QStringLiteral("Breeze"));
     }
 }
-
-QString gtk2Theme()
-{
-    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
-    QFile gtkrc(gtkrcPath);
-    if (gtkrc.open(QIODevice::ReadWrite | QIODevice::Text)) {
-        const QRegularExpression regExp(QStringLiteral("gtk-theme-name=[^\n]*($|\n)"));
-        while (!gtkrc.atEnd()) {
-            QString line = gtkrc.readLine();
-            if (line.contains(regExp)) {
-                return line.split('"')[1];
-            }
-        }
-    }
-
-    return QString();
-}
