diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5beba36..0f4b2c0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,6 @@ find_package(Sass REQUIRED)
 find_package(XSettingsd)
 
 pkg_check_modules(GTK+3 REQUIRED IMPORTED_TARGET gtk+-3.0)
-pkg_check_modules(GTK+2 REQUIRED IMPORTED_TARGET gtk+-2.0)
 pkg_check_modules(GLib2 REQUIRED IMPORTED_TARGET glib-2.0)
 pkg_check_modules(GObject REQUIRED IMPORTED_TARGET gobject-2.0)
 pkg_check_modules(GIO REQUIRED IMPORTED_TARGET gio-2.0)
diff --git a/kconf_update/gtktheme.cpp b/kconf_update/gtktheme.cpp
index 22ee9f0..2a33666 100644
--- a/kconf_update/gtktheme.cpp
+++ b/kconf_update/gtktheme.cpp
@@ -11,27 +11,13 @@
 
 #include "configeditor.h"
 
-QString gtk2Theme();
-void upgradeGtk2Theme();
 void upgradeGtk3Theme();
 
 int main() {
-    upgradeGtk2Theme();
     upgradeGtk3Theme();
     return 0;
 }
 
-void upgradeGtk2Theme() {
-    QString currentGtk2Theme = gtk2Theme();
-    if (currentGtk2Theme.isEmpty()
-     || currentGtk2Theme == QStringLiteral("oxygen-gtk")
-     || currentGtk2Theme == QStringLiteral("BreezyGTK")
-     || currentGtk2Theme == QStringLiteral("Orion")
-    ) {
-        ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-theme-name"), QStringLiteral("Breeze"));
-    }
-}
-
 void upgradeGtk3Theme() {
     QString currentGtk3Theme = ConfigEditor::gtk3ConfigValueSettingsIni(QStringLiteral("gtk-theme-name"));
     if (currentGtk3Theme.isEmpty()
@@ -44,20 +30,3 @@ void upgradeGtk3Theme() {
         ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Net/ThemeName"), QStringLiteral("Breeze"));
     }
 }
-
-QString gtk2Theme()
-{
-    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
-    QFile gtkrc(gtkrcPath);
-    if (gtkrc.open(QIODevice::ReadWrite | QIODevice::Text)) {
-        const QRegularExpression regExp(QStringLiteral("gtk-theme-name=[^\n]*($|\n)"));
-        while (!gtkrc.atEnd()) {
-            QString line = gtkrc.readLine();
-            if (line.contains(regExp)) {
-                return line.split('"')[1];
-            }
-        }
-    }
-
-    return QString();
-}
diff --git a/kded/configeditor.cpp b/kded/configeditor.cpp
index da646c3..1a87743 100644
--- a/kded/configeditor.cpp
+++ b/kded/configeditor.cpp
@@ -78,20 +78,6 @@ void ConfigEditor::setGtk3ConfigValueXSettingsd(const QString &paramName, const
     reloadXSettingsd();
 }
 
-void ConfigEditor::setGtk2ConfigValue(const QString &paramName, const QVariant &paramValue)
-{
-    QString gtkrcPath = qEnvironmentVariable("GTK2_RC_FILES", QDir::homePath() + QStringLiteral("/.gtkrc-2.0"));
-    if (gtkrcPath.contains(QStringLiteral(":/"))) { // I.e. env variable contains multiple paths
-        gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
-    }
-    QFile gtkrc(gtkrcPath);
-    QString gtkrcContents = readFileContents(gtkrc);
-    replaceValueInGtkrcContents(gtkrcContents, paramName, paramValue);
-    gtkrc.remove();
-    gtkrc.open(QIODevice::WriteOnly | QIODevice::Text);
-    gtkrc.write(gtkrcContents.toUtf8());
-}
-
 void ConfigEditor::setCustomClientSideDecorations(const QStringList &windowDecorationsButtonsImages)
 {
     saveWindowDecorationsToAssets(windowDecorationsButtonsImages);
@@ -124,33 +110,6 @@ QString ConfigEditor::gtk3ConfigValueSettingsIni(const QString& paramName)
 }
 
 
-void ConfigEditor::removeLegacyGtk2Strings()
-{
-    QString gtkrcPath = QDir::homePath() + QStringLiteral("/.gtkrc-2.0");
-    QFile gtkrc(gtkrcPath);
-    QString gtkrcContents = readFileContents(gtkrc);
-
-    // Remove "include" lines
-    // Example:
-    // include "/usr/share/themes/Adwaita-dark/gtk-2.0/gtkrc"
-    static const QRegularExpression includeRegExp(QStringLiteral("include .*\n"));
-    gtkrcContents.remove(includeRegExp);
-
-    // Remove redundant font config lines
-    // Example:
-    // style "user-font"
-    // {
-    //     font_name="Noto Sans Regular"
-    // }
-    // widget_class "*" style "user-font"
-    static const QRegularExpression userFontStyleRegexp(QStringLiteral("style(.|\n)*{(.|\n)*}\nwidget_class.*\"user-font\""));
-    gtkrcContents.remove(userFontStyleRegexp);
-
-    gtkrc.remove();
-    gtkrc.open(QIODevice::WriteOnly | QIODevice::Text);
-    gtkrc.write(gtkrcContents.toUtf8());
-}
-
 void ConfigEditor::saveWindowDecorationsToAssets(const QStringList &windowDecorationsButtonsImages)
 {
     QDir assetsFolder {QStandardPaths::writableLocation(QStandardPaths::GenericConfigLocation) + QStringLiteral("/gtk-3.0/assets")};
diff --git a/kded/configeditor.h b/kded/configeditor.h
index e9c5eeb..19faa27 100644
--- a/kded/configeditor.h
+++ b/kded/configeditor.h
@@ -14,8 +14,6 @@ class QVariant;
 
 namespace ConfigEditor
 {
-    void setGtk2ConfigValue(const QString &paramName, const QVariant &paramValue);
-
     void setGtk3ConfigValueGSettings(const QString &paramName, const QVariant &paramValue, const QString &category = QStringLiteral("org.gnome.desktop.interface"));
     void setGtk3ConfigValueGSettingsAsEnum(const QString &paramName, int paramValue, const QString &category = QStringLiteral("org.gnome.desktop.interface"));
     void setGtk3ConfigValueSettingsIni(const QString &paramName, const QVariant &paramValue);
@@ -27,8 +25,6 @@ namespace ConfigEditor
 
     QString gtk3ConfigValueSettingsIni(const QString& paramName);
 
-    void removeLegacyGtk2Strings();
-
 
     void saveWindowDecorationsToAssets(const QStringList &windowDecorationsButtonsImages);
     void addWindowDecorationsCssFile();
diff --git a/kded/gtkconfig.cpp b/kded/gtkconfig.cpp
index 1312997..810034f 100644
--- a/kded/gtkconfig.cpp
+++ b/kded/gtkconfig.cpp
@@ -38,7 +38,6 @@ GtkConfig::GtkConfig(QObject *parent, const QVariantList&) :
     connect(kcminputConfigWatcher.data(), &KConfigWatcher::configChanged, this, &GtkConfig::onKCMInputSettingsChange);
     connect(breezeConfigWatcher.data(), &KConfigWatcher::configChanged, this, &GtkConfig::onBreezeSettingsChange);
 
-    ConfigEditor::removeLegacyGtk2Strings();
     applyAllSettings();
 }
 
@@ -51,7 +50,6 @@ GtkConfig::~GtkConfig()
 
 void GtkConfig::setGtkTheme(const QString &themeName) const
 {
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-theme-name"), themeName);
     ConfigEditor::setGtk3ConfigValueGSettings(QStringLiteral("gtk-theme"), themeName);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-theme-name"), themeName);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Net/ThemeName"),  themeName);
@@ -73,7 +71,6 @@ void GtkConfig::showGtkThemePreview(const QString& themeName) const
 void GtkConfig::setFont() const
 {
     const QString configFontName = configValueProvider->fontName();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-font-name"), configFontName);
     ConfigEditor::setGtk3ConfigValueGSettings(QStringLiteral("font-name"), configFontName);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-font-name"), configFontName);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/FontName"),  configFontName);
@@ -82,7 +79,6 @@ void GtkConfig::setFont() const
 void GtkConfig::setIconTheme() const
 {
     const QString iconThemeName = configValueProvider->iconThemeName();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-icon-theme-name"), iconThemeName);
     ConfigEditor::setGtk3ConfigValueGSettings(QStringLiteral("icon-theme"), iconThemeName);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-icon-theme-name"), iconThemeName);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Net/IconThemeName"),  iconThemeName);
@@ -91,7 +87,6 @@ void GtkConfig::setIconTheme() const
 void GtkConfig::setCursorTheme() const
 {
     const QString cursorThemeName = configValueProvider->cursorThemeName();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-cursor-theme-name"), cursorThemeName);
     ConfigEditor::setGtk3ConfigValueGSettings(QStringLiteral("cursor-theme"), cursorThemeName);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-cursor-theme-name"), cursorThemeName);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/CursorThemeName"),  cursorThemeName);
@@ -100,7 +95,6 @@ void GtkConfig::setCursorTheme() const
 void GtkConfig::setCursorSize() const
 {
     const int cursorSize = configValueProvider->cursorSize();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-cursor-theme-size"), cursorSize);
     ConfigEditor::setGtk3ConfigValueGSettings(QStringLiteral("cursor-size"), cursorSize);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-cursor-theme-size"), cursorSize);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/CursorThemeSize"),  cursorSize);
@@ -109,7 +103,6 @@ void GtkConfig::setCursorSize() const
 void GtkConfig::setIconsOnButtons() const
 {
     const bool iconsOnButtonsConfigValue = configValueProvider->iconsOnButtons();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-button-images"), iconsOnButtonsConfigValue);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-button-images"), iconsOnButtonsConfigValue);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/ButtonImages"), iconsOnButtonsConfigValue);
 }
@@ -117,7 +110,6 @@ void GtkConfig::setIconsOnButtons() const
 void GtkConfig::setIconsInMenus() const
 {
     const bool iconsInMenusConfigValue = configValueProvider->iconsInMenus();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-menu-images"), iconsInMenusConfigValue);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-menu-images"), iconsInMenusConfigValue);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/MenuImages"), iconsInMenusConfigValue);
 }
@@ -125,7 +117,6 @@ void GtkConfig::setIconsInMenus() const
 void GtkConfig::setToolbarStyle() const
 {
     const int toolbarStyle = configValueProvider->toolbarStyle();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-toolbar-style"), toolbarStyle);
     ConfigEditor::setGtk3ConfigValueGSettingsAsEnum(QStringLiteral("toolbar-style"), toolbarStyle);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-toolbar-style"), toolbarStyle);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/ToolbarStyle"), toolbarStyle);
@@ -134,7 +125,6 @@ void GtkConfig::setToolbarStyle() const
 void GtkConfig::setScrollbarBehavior() const
 {
     const bool scrollbarBehavior = configValueProvider->scrollbarBehavior();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-primary-button-warps-slider"), scrollbarBehavior);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-primary-button-warps-slider"), scrollbarBehavior);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/PrimaryButtonWarpsSlider"), scrollbarBehavior);
 }
@@ -166,7 +156,6 @@ void GtkConfig::setWindowDecorationsButtonsOrder() const
 void GtkConfig::setEnableAnimations() const
 {
     const bool enableAnimations = configValueProvider->enableAnimations();
-    ConfigEditor::setGtk2ConfigValue(QStringLiteral("gtk-enable-animations"), enableAnimations);
     ConfigEditor::setGtk3ConfigValueGSettings(QStringLiteral("enable-animations"), enableAnimations);
     ConfigEditor::setGtk3ConfigValueSettingsIni(QStringLiteral("gtk-enable-animations"), enableAnimations);
     ConfigEditor::setGtk3ConfigValueXSettingsd(QStringLiteral("Gtk/EnableAnimations"), enableAnimations);
