diff --git a/src/widgets/ldapclient.cpp b/src/widgets/ldapclient.cpp
index a419458..1c7391a 100644
--- a/src/widgets/ldapclient.cpp
+++ b/src/widgets/ldapclient.cpp
@@ -31,7 +31,6 @@ public:
 
     ~LdapClientPrivate()
     {
-        cancelQuery();
     }
 
     void cancelQuery();
@@ -45,7 +44,7 @@ public:
     void slotInfoMessage(KJob *, const QString &info, const QString &);
     void slotDone();
 
-    LdapClient *const q;
+    LdapClient *const q = nullptr;
 
     KLDAP::LdapServer mServer;
     QString mScope;
@@ -68,7 +67,7 @@ LdapClient::LdapClient(int clientNumber, QObject *parent)
     d->mCompletionWeight = 50 - d->mClientNumber;
 }
 
-LdapClient::~LdapClient() = default;
+LdapClient::~LdapClient() { cancelQuery(); };
 
 bool LdapClient::isActive() const
 {
