diff --git a/src/modules/bluetooth/backend-ofono.c b/src/modules/bluetooth/backend-ofono.c
index 85dec44..7698830 100644
--- a/src/modules/bluetooth/backend-ofono.c
+++ b/src/modules/bluetooth/backend-ofono.c
@@ -518,8 +518,6 @@ static void hf_audio_agent_register_reply(DBusPendingCall *pending, void *userda
     pa_assert_se(r = dbus_pending_call_steal_reply(pending));
 
     if (dbus_message_get_type(r) == DBUS_MESSAGE_TYPE_ERROR) {
-        pa_log_info("Failed to register as a handsfree audio agent with ofono: %s: %s",
-                    dbus_message_get_error_name(r), pa_dbus_get_error_message(r));
         goto finish;
     }
 
