diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index 64e9f21..e9be0c9 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -1841,12 +1841,6 @@ char *pa_get_runtime_dir(void) {
     /* Use the explicitly configured value if it is set */
     d = getenv("PULSE_RUNTIME_PATH");
     if (d) {
-
-        if (pa_make_secure_dir(d, m, (uid_t) -1, (gid_t) -1, true) < 0) {
-            pa_log_error("Failed to create secure directory (%s): %s", d, pa_cstrerror(errno));
-            goto fail;
-        }
-
         return pa_xstrdup(d);
     }
 
