diff --git a/builtin/commit.c b/builtin/commit.c
index 190d215..a6636f4 100644
--- a/builtin/commit.c
+++ b/builtin/commit.c
@@ -1767,17 +1767,6 @@ int cmd_commit(int argc, const char **argv, const char *prefix)
 
 	cleanup_message(&sb, cleanup_mode, verbose);
 
-	if (message_is_empty(&sb, cleanup_mode) && !allow_empty_message) {
-		rollback_index_files();
-		fprintf(stderr, _("Aborting commit due to empty commit message.\n"));
-		exit(1);
-	}
-	if (template_untouched(&sb, template_file, cleanup_mode) && !allow_empty_message) {
-		rollback_index_files();
-		fprintf(stderr, _("Aborting commit; you did not edit the message.\n"));
-		exit(1);
-	}
-
 	if (fixup_message && starts_with(sb.buf, "amend! ") &&
 	    !allow_empty_message) {
 		struct strbuf body = STRBUF_INIT;
