diff --git a/matcher/multimatch.cc b/matcher/multimatch.cc
index 282ec16..8654f81 100644
--- a/matcher/multimatch.cc
+++ b/matcher/multimatch.cc
@@ -453,6 +453,7 @@ MultiMatch::get_mset(Xapian::doccount first, Xapian::doccount maxitems,
 	    Xapian::termcount total_subqs_i = 0;
 	    PostList* pl = leaves[i]->get_postlist(this, &total_subqs_i);
 	    total_subqs = max(total_subqs, total_subqs_i);
+#ifdef XAPIAN_HAS_REMOTE_BACKEND
 	    if (is_remote[i]) {
 		++n_remotes;
 		RemoteSubMatch* rem_match =
@@ -469,6 +470,7 @@ MultiMatch::get_mset(Xapian::doccount first, Xapian::doccount maxitems,
 		    definite_matches_not_seen -= first + maxitems;
 		}
 	    }
+#endif
 	    postlists.push_back(pl);
 	}
     } catch (...) {
