diff --git a/src/grantleethememanager.cpp b/src/grantleethememanager.cpp
index 879593b..6a847f8 100644
--- a/src/grantleethememanager.cpp
+++ b/src/grantleethememanager.cpp
@@ -106,6 +106,9 @@ public:
             while (dirIt.hasNext()) {
                 dirIt.next();
                 const QString dirName = dirIt.fileName();
+                if (dirName.isEmpty() || dirIt.filePath().isEmpty()) {
+                    continue;
+                }
                 GrantleeTheme::Theme theme = q->loadTheme(dirIt.filePath(), dirName, defaultDesktopFileName);
                 if (theme.isValid()) {
                     QString themeName = theme.name();
