diff --git a/src/grantleethememanager.cpp b/src/grantleethememanager.cpp
index 3db9a62..2c2c208 100644
--- a/src/grantleethememanager.cpp
+++ b/src/grantleethememanager.cpp
@@ -130,6 +130,9 @@ public:
             while (dirIt.hasNext()) {
                 dirIt.next();
                 const QString dirName = dirIt.fileName();
+                if (dirName.isEmpty() || dirIt.filePath().isEmpty()) {
+                    continue;
+                }
                 GrantleeTheme::Theme theme = q->loadTheme(dirIt.filePath(), dirName, defaultDesktopFileName);
                 if (theme.isValid()) {
                     QString themeName = theme.name();
