diff --git a/src/modules/generic.c b/src/modules/generic.c
index f788c40..e3b201c 100644
--- a/src/modules/generic.c
+++ b/src/modules/generic.c
@@ -219,9 +219,9 @@ int module_speak(gchar * data, size_t bytes, SPDMessageType msgtype)
 						    GenericRecodeFallback, NULL,
 						    NULL, NULL);
 	} else {
-		DBG("Warning: Preferred charset not specified, recoding to iso-8859-1");
+		DBG("Warning: Preferred charset not specified, recoding to UTF-8");
 		tmp =
-		    (char *)g_convert_with_fallback(data, bytes, "iso-8859-2",
+		    (char *)g_convert_with_fallback(data, bytes, "UTF-8",
 						    "UTF-8",
 						    GenericRecodeFallback, NULL,
 						    NULL, NULL);
