diff --git a/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro b/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro
index 4d5835d..ffdef3e 100644
--- a/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro
+++ b/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro
@@ -2,7 +2,8 @@ TARGET = qmapboxgl
 
 load(qt_helper_lib)
 
-CONFIG += qt c++17 exceptions warn_off staticlib object_parallel_to_source
+CONFIG += qt c++14 exceptions warn_off staticlib object_parallel_to_source
+CONFIG -= c++1z c++17
 
 QT += network-private \
       gui-private \
