diff --git a/Modules/about-distro/src/about-distro.desktop b/Modules/about-distro/src/about-distro.desktop
index 0890ced..dfdc489 100644
--- a/Modules/about-distro/src/about-distro.desktop
+++ b/Modules/about-distro/src/about-distro.desktop
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Type=Service
 Icon=ktip
-Exec=kinfocenter about-distro
+Exec=kcmshell5 about-distro
 
 X-KDE-Library=kcm_about_distro
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/devinfo/devinfo.desktop b/Modules/devinfo/devinfo.desktop
index d7196bd..9f692f1 100644
--- a/Modules/devinfo/devinfo.desktop
+++ b/Modules/devinfo/devinfo.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter devinfo
+Exec=kcmshell5 devinfo
 Icon=device-notifier
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/energy/kcm_energyinfo.desktop b/Modules/energy/kcm_energyinfo.desktop
index ca3c18a..48c748d 100644
--- a/Modules/energy/kcm_energyinfo.desktop
+++ b/Modules/energy/kcm_energyinfo.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter kcm_energyinfo
+Exec=kcmshell5 kcm_energyinfo
 Icon=battery
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/fileindexermonitor/kcm_fileindexermonitor.desktop b/Modules/fileindexermonitor/kcm_fileindexermonitor.desktop
index 192d92c..535dabf 100644
--- a/Modules/fileindexermonitor/kcm_fileindexermonitor.desktop
+++ b/Modules/fileindexermonitor/kcm_fileindexermonitor.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter kcm_fileindexermonitor
+Exec=kcmshell5 kcm_fileindexermonitor
 Icon=baloo
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/info/dma.desktop b/Modules/info/dma.desktop
index 8d2bf64..6b7c587 100644
--- a/Modules/info/dma.desktop
+++ b/Modules/info/dma.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter dma
+Exec=kcmshell5 dma
 Icon=media-flash
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/info/interrupts.desktop b/Modules/info/interrupts.desktop
index 32003ad..34a9dcb 100644
--- a/Modules/info/interrupts.desktop
+++ b/Modules/info/interrupts.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter interrupts
+Exec=kcmshell5 interrupts
 Icon=cpu
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/info/ioports.desktop b/Modules/info/ioports.desktop
index e56785f..05ac18a 100644
--- a/Modules/info/ioports.desktop
+++ b/Modules/info/ioports.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter ioports
+Exec=kcmshell5 ioports
 Icon=computer
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/info/wayland.desktop b/Modules/info/wayland.desktop
index 25c83e9..dd6bbf7 100644
--- a/Modules/info/wayland.desktop
+++ b/Modules/info/wayland.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter wayland
+Exec=kcmshell5 wayland
 Icon=wayland
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/info/xserver.desktop b/Modules/info/xserver.desktop
index e12e24b..cbf6430 100644
--- a/Modules/info/xserver.desktop
+++ b/Modules/info/xserver.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter xserver
+Exec=kcmshell5 xserver
 Icon=xorg
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/memory/kcm_memory.desktop b/Modules/memory/kcm_memory.desktop
index 574317f..ba2a683 100644
--- a/Modules/memory/kcm_memory.desktop
+++ b/Modules/memory/kcm_memory.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter kcm_memory
+Exec=kcmshell5 kcm_memory
 Icon=partitionmanager
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/nics/nic.desktop b/Modules/nics/nic.desktop
index 88e33c1..ea65fa8 100644
--- a/Modules/nics/nic.desktop
+++ b/Modules/nics/nic.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter nic
+Exec=kcmshell5 nic
 Type=Service
 X-KDE-ServiceTypes=KCModule
 Icon=network-card
diff --git a/Modules/opengl/opengl.desktop b/Modules/opengl/opengl.desktop
index 0016f6b..d20c5d9 100644
--- a/Modules/opengl/opengl.desktop
+++ b/Modules/opengl/opengl.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter opengl
+Exec=kcmshell5 opengl
 Type=Service
 X-KDE-ServiceTypes=KCModule
 Icon=video-display
diff --git a/Modules/pci/kcm_pci.desktop b/Modules/pci/kcm_pci.desktop
index 804d532..0e183f5 100644
--- a/Modules/pci/kcm_pci.desktop
+++ b/Modules/pci/kcm_pci.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter kcm_pci
+Exec=kcmshell5 kcm_pci
 Icon=hwinfo
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/samba/smbstatus.desktop b/Modules/samba/smbstatus.desktop
index ecf64b3..77117c4 100644
--- a/Modules/samba/smbstatus.desktop
+++ b/Modules/samba/smbstatus.desktop
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Exec=kinfocenter smbstatus
+Exec=kcmshell5 smbstatus
 Icon=preferences-system-network-sharing
 Type=Service
 X-KDE-ServiceTypes=KCModule
diff --git a/Modules/usbview/kcmusb.desktop b/Modules/usbview/kcmusb.desktop
index 4840674..280666e 100644
--- a/Modules/usbview/kcmusb.desktop
+++ b/Modules/usbview/kcmusb.desktop
@@ -2,7 +2,7 @@
 Icon=drive-removable-media-usb-pendrive
 Type=Service
 X-KDE-ServiceTypes=KCModule
-Exec=kinfocenter kcmusb
+Exec=kcmshell5 kcmusb
 
 X-KDE-KInfoCenter-Category=device_information
 X-KDE-Library=kcm_usb
diff --git a/Modules/view1394/kcmview1394.desktop b/Modules/view1394/kcmview1394.desktop
index b772aac..c837689 100644
--- a/Modules/view1394/kcmview1394.desktop
+++ b/Modules/view1394/kcmview1394.desktop
@@ -2,7 +2,7 @@
 Icon=camera-web
 Type=Service
 X-KDE-ServiceTypes=KCModule
-Exec=kinfocenter kcmview1394
+Exec=kcmshell5 kcmview1394
 
 X-KDE-KInfoCenter-Category=device_information
 X-KDE-Library=kcm_view1394
