diff --git a/polkit-kde-authentication-agent-1.desktop.cmake b/polkit-kde-authentication-agent-1.desktop.cmake
index 6d7dd1e..e511622 100644
--- a/polkit-kde-authentication-agent-1.desktop.cmake
+++ b/polkit-kde-authentication-agent-1.desktop.cmake
@@ -111,7 +111,7 @@ Comment[vi]=Đại diện xác thực PolicyKit
 Comment[x-test]=xxPolicyKit Authentication Agentxx
 Comment[zh_CN]=PolicyKit 身份验证代理程序
 Comment[zh_TW]=PolicyKit 認證代理程式
-Exec=${KDE_INSTALL_FULL_LIBEXECDIR}/polkit-kde-authentication-agent-1
+Exec=/usr/libexec/polkit-kde-authentication-agent-1
 Icon=dialog-password
 Terminal=false
 Type=Application
