diff --git a/lib/Driver/ToolChains/Linux.cpp b/lib/Driver/ToolChains/Linux.cpp
index 1ba222bf83b1..b123de5f2ad1 100644
--- a/lib/Driver/ToolChains/Linux.cpp
+++ b/lib/Driver/ToolChains/Linux.cpp
@@ -337,9 +337,11 @@ Linux::Linux(const Driver &D, const llvm::Triple &Triple, const ArgList &Args)
 }
 
 ToolChain::RuntimeLibType Linux::GetDefaultRuntimeLibType() const {
-  if (getTriple().isAndroid())
-    return ToolChain::RLT_CompilerRT;
-  return Generic_ELF::GetDefaultRuntimeLibType();
+  return ToolChain::RLT_CompilerRT;
+}
+
+ToolChain::UnwindLibType Linux::GetDefaultUnwindLibType() const {
+  return ToolChain::UNW_CompilerRT;
 }
 
 unsigned Linux::GetDefaultDwarfVersion() const {
@@ -349,9 +351,7 @@ unsigned Linux::GetDefaultDwarfVersion() const {
 }
 
 ToolChain::CXXStdlibType Linux::GetDefaultCXXStdlibType() const {
-  if (getTriple().isAndroid())
-    return ToolChain::CST_Libcxx;
-  return ToolChain::CST_Libstdcxx;
+  return ToolChain::CST_Libcxx;
 }
 
 bool Linux::HasNativeLLVMSupport() const { return true; }
diff --git a/lib/Driver/ToolChains/Linux.h b/lib/Driver/ToolChains/Linux.h
index 524391743090..bd3349f7dda5 100644
--- a/lib/Driver/ToolChains/Linux.h
+++ b/lib/Driver/ToolChains/Linux.h
@@ -42,6 +42,7 @@ public:
   void AddIAMCUIncludeArgs(const llvm::opt::ArgList &DriverArgs,
                            llvm::opt::ArgStringList &CC1Args) const override;
   RuntimeLibType GetDefaultRuntimeLibType() const override;
+  UnwindLibType GetDefaultUnwindLibType() const override;
   unsigned GetDefaultDwarfVersion() const override;
   CXXStdlibType GetDefaultCXXStdlibType() const override;
   bool
