diff --git a/lib/Basic/LangStandards.cpp b/lib/Basic/LangStandards.cpp
index af9cf4f27392..c1891bd4e7db 100644
--- a/lib/Basic/LangStandards.cpp
+++ b/lib/Basic/LangStandards.cpp
@@ -59,14 +59,14 @@ LangStandard::Kind clang::getDefaultLanguageStandard(clang::Language Lang,
     // The PS4 uses C99 as the default C standard.
     if (T.isPS4())
       return LangStandard::lang_gnu99;
-    return LangStandard::lang_gnu17;
+    return LangStandard::lang_c17;
   case Language::ObjC:
-    return LangStandard::lang_gnu11;
+    return LangStandard::lang_c11;
   case Language::CXX:
   case Language::ObjCXX:
   case Language::CUDA:
   case Language::HIP:
-    return LangStandard::lang_gnucxx17;
+    return LangStandard::lang_cxx17;
   case Language::RenderScript:
     return LangStandard::lang_c99;
   case Language::HLSL:
