diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb4b801f01c8..c73687e64973 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -269,9 +269,9 @@ set(CLANG_SPAWN_CC1 OFF CACHE BOOL
     "Whether clang should use a new process for the CC1 invocation")
 
 # TODO: verify the values against LangStandards.def?
-set(CLANG_DEFAULT_STD_C "" CACHE STRING
+set(CLANG_DEFAULT_STD_C "c17" CACHE STRING
   "Default standard to use for C/ObjC code (IDENT from LangStandards.def, empty for platform default)")
-set(CLANG_DEFAULT_STD_CXX "" CACHE STRING
+set(CLANG_DEFAULT_STD_CXX "cxx17" CACHE STRING
   "Default standard to use for C++/ObjC++ code (IDENT from LangStandards.def, empty for platform default)")
 
 set(CLANG_DEFAULT_LINKER "" CACHE STRING
diff --git a/include/clang/Config/config.h.cmake b/include/clang/Config/config.h.cmake
index 26e9d5c4eb4d..02b546e02bdd 100644
--- a/include/clang/Config/config.h.cmake
+++ b/include/clang/Config/config.h.cmake
@@ -12,10 +12,10 @@
 #define CLANG_DEFAULT_LINKER "${CLANG_DEFAULT_LINKER}"
 
 /* Default C/ObjC standard to use. */
-#cmakedefine CLANG_DEFAULT_STD_C LangStandard::lang_${CLANG_DEFAULT_STD_C}
+#cmakedefine CLANG_DEFAULT_STD_C LangStandard::lang_c17
 
 /* Default C++/ObjC++ standard to use. */
-#cmakedefine CLANG_DEFAULT_STD_CXX LangStandard::lang_${CLANG_DEFAULT_STD_CXX}
+#cmakedefine CLANG_DEFAULT_STD_CXX LangStandard::lang_cxx17
 
 /* Default C++ stdlib to use. */
 #define CLANG_DEFAULT_CXX_STDLIB "${CLANG_DEFAULT_CXX_STDLIB}"
