diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index cfaa253..83f3f58 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -172,7 +172,7 @@ if(MINGW)
     set_source_files_properties(utils/gnupg-registry.c PROPERTIES COMPILE_OPTIONS "-Wno-pointer-sign")
 endif()
 
-target_link_libraries(KF5Libkleo PUBLIC Gpgmepp PRIVATE Qt::Widgets
+target_link_libraries(KF5Libkleo PUBLIC Gpgmepp "-lgpg-error" PRIVATE Qt::Widgets
                                                 KF5::I18n
                                                 KF5::Completion
                                                 KF5::ConfigCore
