diff --git a/meson.build b/meson.build
index fc216d2..c494195 100644
--- a/meson.build
+++ b/meson.build
@@ -1329,7 +1329,7 @@ else
 endif
 conf.set10('ENABLE_IMPORTD', have)
 
-want_homed = get_option('homed')
+want_homed = 'false'
 if want_homed != 'false'
         have = (conf.get('HAVE_OPENSSL') == 1 and
                 conf.get('HAVE_LIBFDISK') == 1 and
@@ -1340,7 +1340,7 @@ if want_homed != 'false'
 else
         have = false
 endif
-conf.set10('ENABLE_HOMED', have)
+conf.set10('ENABLE_HOMED', false)
 
 have = have and conf.get('HAVE_PAM') == 1
 conf.set10('ENABLE_PAM_HOME', have)
@@ -2064,7 +2064,7 @@ if conf.get('ENABLE_USERDB') == 1
                    install_dir : rootbindir)
 endif
 
-if conf.get('ENABLE_HOMED') == 1
+if 0 == 1
         executable('systemd-homework',
                    systemd_homework_sources,
                    include_directories : includes,
