diff --git a/configure b/configure
index a3cc8e7a9..c2239df86 100755
--- a/configure
+++ b/configure
@@ -5840,7 +5840,7 @@ $as_echo "ok ($host_cpu)" >&6; }
     fi
     ;;
 
-linux-gnu*|k*bsd*-gnu*)
+linux-gnu*|k*bsd*-gnu*|linux|*linux*)
     test_gtk=yes
     build_gstreamer_1_0=yes
     build_gstreamer_0_10=yes
@@ -6053,7 +6053,22 @@ haiku*)
     ;;
 
 *)
-    as_fn_error $? "$host_os operating system is not suitable to build LibreOffice for!" "$LINENO" 5
+    test_gtk=yes
+    build_gstreamer_1_0=yes
+    build_gstreamer_0_10=yes
+    test_kde5=yes
+    test_gtk3_kde5=yes
+    test_gdb_index=yes
+    test_split_debug=yes
+    if test "$enable_fuzzers" != yes; then
+        test_freetype=yes
+        test_fontconfig=yes
+    else
+        test_freetype=no
+        test_fontconfig=no
+        BUILD_TYPE="$BUILD_TYPE FONTCONFIG FREETYPE"
+    fi
+    _os=Linux
     ;;
 esac
 
@@ -12234,64 +12249,8 @@ $as_echo "$as_me: WARNING: Compiler is not capable of creating split debug info,
 fi
 
 
-ENABLE_GDB_INDEX=
-if test "$enable_gdb_index" != "no"; then
-        if test "$enable_gdb_index" = yes -o \( "$test_gdb_index" = "yes" -o -n "$ENABLE_DEBUG$ENABLE_DBGUTIL" \); then
-        { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC supports -ggnu-pubnames" >&5
-$as_echo_n "checking whether $CC supports -ggnu-pubnames... " >&6; }
-        save_CFLAGS=$CFLAGS
-        CFLAGS="$CFLAGS -Werror -ggnu-pubnames"
-        have_ggnu_pubnames=
-        cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-int
-main ()
-{
- return 0;
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  have_ggnu_pubnames=TRUE
-else
-  have_ggnu_pubnames=
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-        if test "$have_ggnu_pubnames" != "TRUE"; then
-            if test "$enable_gdb_index" = "yes"; then
-                as_fn_error --enable-gdb-index not supported  "no" "$LINENO" 5
-            else
-                { $as_echo "$as_me:${as_lineno-$LINENO}: result: no " >&5
-$as_echo "no " >&6; }
-            fi
-        else
-            { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes " >&5
-$as_echo "yes " >&6; }
-            { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether $CC supports -Wl,--gdb-index" >&5
-$as_echo_n "checking whether $CC supports -Wl,--gdb-index... " >&6; }
-            ldflags_save=$LDFLAGS
-            LDFLAGS="$LDFLAGS -Wl,--gdb-index"
-            cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-#include <stdio.h>
-
-int
-main ()
-{
-
-printf ("hello world\n");
+ENABLE_GDB_INDEX=FALSE
 
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  ENABLE_GDB_INDEX=TRUE
-fi
 rm -f core conftest.err conftest.$ac_objext \
     conftest$ac_exeext conftest.$ac_ext
             if test "$ENABLE_GDB_INDEX" = "TRUE"; then
@@ -12565,8 +12524,8 @@ kfreebsd*)
     esac
     ;;
 
-linux-gnu*)
-    COM=GCC
+linux-gnu*|linux)
+    COM=GCC
     USING_X11=TRUE
     OS=LINUX
     RTL_OS=Linux
diff --git a/configure.ac b/configure.ac
index 0b6497997..0fbec5a08 100644
--- a/configure.ac
+++ b/configure.ac
@@ -601,7 +601,7 @@ solaris*)
     fi
     ;;
 
-linux-gnu*|k*bsd*-gnu*)
+linux-gnu*|k*bsd*-gnu*|linux|*linux*)
     test_gtk=yes
     build_gstreamer_1_0=yes
     build_gstreamer_0_10=yes
@@ -808,7 +808,22 @@ haiku*)
     ;;
 
 *)
-    AC_MSG_ERROR([$host_os operating system is not suitable to build LibreOffice for!])
+    test_gtk=yes
+    build_gstreamer_1_0=yes
+    build_gstreamer_0_10=yes
+    test_kde5=yes
+    test_gtk3_kde5=yes
+    test_gdb_index=yes
+    test_split_debug=yes
+    if test "$enable_fuzzers" != yes; then
+        test_freetype=yes
+        test_fontconfig=yes
+    else
+        test_freetype=no
+        test_fontconfig=no
+        BUILD_TYPE="$BUILD_TYPE FONTCONFIG FREETYPE"
+    fi
+    _os=Linux
     ;;
 esac
 
@@ -4048,50 +4063,7 @@ if test "$enable_split_debug" != no; then
 fi
 AC_SUBST(HAVE_GCC_SPLIT_DWARF)
 
-ENABLE_GDB_INDEX=
-if test "$enable_gdb_index" != "no"; then
-    dnl Currently by default enabled only on Linux, feel free to set test_gdb_index above also for other platforms.
-    if test "$enable_gdb_index" = yes -o \( "$test_gdb_index" = "yes" -o -n "$ENABLE_DEBUG$ENABLE_DBGUTIL" \); then
-        AC_MSG_CHECKING([whether $CC supports -ggnu-pubnames])
-        save_CFLAGS=$CFLAGS
-        CFLAGS="$CFLAGS -Werror -ggnu-pubnames"
-        have_ggnu_pubnames=
-        AC_LINK_IFELSE([AC_LANG_PROGRAM([[]], [[ return 0; ]])],[have_ggnu_pubnames=TRUE],[have_ggnu_pubnames=])
-        if test "$have_ggnu_pubnames" != "TRUE"; then
-            if test "$enable_gdb_index" = "yes"; then
-                AC_MSG_ERROR( no, --enable-gdb-index not supported )
-            else
-                AC_MSG_RESULT( no )
-            fi
-        else
-            AC_MSG_RESULT( yes )
-            AC_MSG_CHECKING([whether $CC supports -Wl,--gdb-index])
-            ldflags_save=$LDFLAGS
-            LDFLAGS="$LDFLAGS -Wl,--gdb-index"
-            AC_LINK_IFELSE([AC_LANG_PROGRAM([
-#include <stdio.h>
-                ],[
-printf ("hello world\n");
-                ])], ENABLE_GDB_INDEX=TRUE, [])
-            if test "$ENABLE_GDB_INDEX" = "TRUE"; then
-                AC_MSG_RESULT( yes )
-            else
-                if test "$enable_gdb_index" = "yes"; then
-                    AC_MSG_ERROR( no )
-                else
-                    AC_MSG_RESULT( no )
-                fi
-            fi
-            LDFLAGS=$ldflags_save
-        fi
-        CFLAGS=$save_CFLAGS
-        fi
-    if test -z "$ENABLE_GDB_INDEX" -a "$test_gdb_index" = "yes" -a -n "$ENABLE_DEBUG$ENABLE_DBGUTIL"; then
-        AC_MSG_WARN([Linker is not capable of creating gdb index, debugger startup will be slow.])
-        add_warning "Linker is not capable of creating gdb index, debugger startup will be slow."
-    fi
-fi
-AC_SUBST(ENABLE_GDB_INDEX)
+ENABLE_GDB_INDEX=FALSE
 
 if test "$enable_sal_log" = yes; then
     ENABLE_SAL_LOG=TRUE
@@ -4334,8 +4306,8 @@ kfreebsd*)
     esac
     ;;
 
-linux-gnu*)
-    COM=GCC
+linux-gnu*|linux)
+    COM=GCC
     USING_X11=TRUE
     OS=LINUX
     RTL_OS=Linux
