diff --git a/Source/cmake.cxx b/Source/cmake.cxx
index a99d9a6..86e6ad1 100644
--- a/Source/cmake.cxx
+++ b/Source/cmake.cxx
@@ -1718,7 +1718,7 @@ std::unique_ptr<cmGlobalGenerator> cmake::EvaluateDefaultGlobalGenerator()
   return std::unique_ptr<cmGlobalGenerator>(std::move(gen));
 #else
   return std::unique_ptr<cmGlobalGenerator>(
-    cm::make_unique<cmGlobalUnixMakefileGenerator3>(this));
+    cm::make_unique<cmGlobalNinjaGenerator>(this));
 #endif
 }
 
