diff --git a/applets/pager/package/contents/ui/main.qml b/applets/pager/package/contents/ui/main.qml
index 764026a..36c25f5 100644
--- a/applets/pager/package/contents/ui/main.qml
+++ b/applets/pager/package/contents/ui/main.qml
@@ -477,7 +477,7 @@ PlasmoidItem {
                             y: Math.round(geometry.y * pagerItemGrid.heightScaleFactor) - 1
                             width: Math.round(geometry.width * pagerItemGrid.widthScaleFactor)
                             height: Math.round(geometry.height * pagerItemGrid.heightScaleFactor)
-                            visible: Plasmoid.configuration.showWindowOutlines && !model.IsMinimized
+                            visible: true
                             color: {
                                 if (desktop.active) {
                                     if (model.IsActive) {
