diff --git a/applets/pager/package/contents/ui/main.qml b/applets/pager/package/contents/ui/main.qml
index 4881f67..b73ff03 100644
--- a/applets/pager/package/contents/ui/main.qml
+++ b/applets/pager/package/contents/ui/main.qml
@@ -437,7 +437,7 @@ MouseArea {
                             y: (geometry.y * pagerItemGrid.heightScaleFactor) - Math.round(PlasmaCore.Units.devicePixelRatio)
                             width: geometry.width * pagerItemGrid.widthScaleFactor
                             height: geometry.height * pagerItemGrid.heightScaleFactor
-                            visible: model.IsMinimized !== true
+                            visible: true
                             color: {
                                 if (desktop.active) {
                                     if (model.IsActive === true)
