diff --git a/applets/taskmanager/package/contents/ui/ContextMenu.qml b/applets/taskmanager/package/contents/ui/ContextMenu.qml
index 5a87986..ae7675b 100644
--- a/applets/taskmanager/package/contents/ui/ContextMenu.qml
+++ b/applets/taskmanager/package/contents/ui/ContextMenu.qml
@@ -91,7 +91,7 @@ PlasmaComponents.ContextMenu {
             {
                 title:   i18n("Places"),
                 group:   "places",
-                actions: backend.placesActions(launcherUrl, showAllPlaces, menu)
+                actions: backend.placesActions(launcherUrl, true, menu)
             },
             {
                 title:   i18n("Recent Files"),
