diff --git a/Makefile b/Makefile
index 08d2570..a7d789a 100644
--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,6 @@ endif
 ifeq ($(GOLANG),yes)
 	$(MAKE) -C go $@
 endif
-	$(MAKE) -C tests $@
 	$(MAKE) -C progs $@
 	$(MAKE) -C doc $@
 	$(MAKE) -C kdebug $@
@@ -35,7 +34,6 @@ release: distclean
 	cd .. && ln -s libcap libcap-$(VERSION).$(MINOR) && tar cvf libcap-$(VERSION).$(MINOR).tar --exclude patches libcap-$(VERSION).$(MINOR)/* && rm libcap-$(VERSION).$(MINOR)
 
 test: all
-	cd progs && sudo ./quicktest.sh
 
 morganrelease: distclean
 	@echo "sign the tag twice: older DSA key; and newer RSA kernel.org key"
