diff --git a/config.toml b/config.toml
index 6c1804c..e28a21b 100644
--- a/config.toml
+++ b/config.toml
@@ -45,18 +45,18 @@ optimize-tests = true
 codegen-tests = true
 dist-src = false
 remap-debuginfo = true
-lld = false
+lld = true
 backtrace-on-ice = true
 jemalloc = false
 [dist]
 src-tarball = false
 [target.i686-unknown-linux-gnu]
-cc = "clang"
-cxx = "clang++"
+cc = "/etc/portage/compiler-wrapper/rust-clang"
+cxx = "/etc/portage/compiler-wrapper/rust-clang++"
 linker = "/etc/portage/compiler-wrapper/rust-clang"
 ar = "llvm-ar"
 [target.x86_64-unknown-linux-gnu]
-cc = "clang"
-cxx = "clang++"
+cc = "/etc/portage/compiler-wrapper/rust-clang"
+cxx = "/etc/portage/compiler-wrapper/rust-clang++"
 linker = "/etc/portage/compiler-wrapper/rust-clang"
 ar = "llvm-ar"
