diff --git a/shell/shell.cpp b/shell/shell.cpp
index 15485c0..8519639 100644
--- a/shell/shell.cpp
+++ b/shell/shell.cpp
@@ -258,7 +258,7 @@ bool Shell::openDocument(const QUrl &url, const QString &serializedOptions)
     KParts::ReadWritePart *const part = m_tabs[0].part;
 
     // Return false if we can't open new tabs and the only part is occupied
-    if (!qobject_cast<Okular::ViewerInterface *>(part)->openNewFilesInTabs() && !part->url().isEmpty() && !ShellUtils::unique(serializedOptions)) {
+    if (!part->url().isEmpty() && !ShellUtils::unique(serializedOptions)) {
         return false;
     }
 
@@ -279,7 +279,7 @@ bool Shell::canOpenDocs(int numDocs, int desktop)
     }
 
     KParts::ReadWritePart *const part = m_tabs[0].part;
-    const bool allowTabs = qobject_cast<Okular::ViewerInterface *>(part)->openNewFilesInTabs();
+    const bool allowTabs = false;
 
     if (!allowTabs && (numDocs > 1 || !part->url().isEmpty())) {
         return false;
