diff --git a/package/contents/config/config.qml b/package/contents/config/config.qml
index 966545e..b8cadb9 100644
--- a/package/contents/config/config.qml
+++ b/package/contents/config/config.qml
@@ -4,7 +4,7 @@ import org.kde.plasma.configuration 2.0
 ConfigModel {
     ConfigCategory {
          name: i18n('General')
-         icon: 'folder-favorites'
+         icon: 'folder'
          source: 'config/ConfigGeneral.qml'
     }
 }
diff --git a/package/contents/ui/main.qml b/package/contents/ui/main.qml
index 895e3d9..bee9aa3 100644
--- a/package/contents/ui/main.qml
+++ b/package/contents/ui/main.qml
@@ -26,7 +26,7 @@ Item {
     property int widgetWidth: plasmoid.configuration.widgetWidth
 
     Plasmoid.compactRepresentation: PlasmaCore.IconItem {
-        source: 'folder-favorites'
+        source: 'folder'
         width: units.iconSizes.medium
         height: units.iconSizes.medium
         active: mouseArea.containsMouse
diff --git a/package/metadata.desktop b/package/metadata.desktop
index 6fbdad9..32e7549 100644
--- a/package/metadata.desktop
+++ b/package/metadata.desktop
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=Places Widget
+Name[ja]=場所
 Comment=Gives access to user places
+Comment[ja]=ユーザーの場所
 Type=Service
 X-KDE-ParentApp=
 X-KDE-PluginInfo-Author=Daniel Faust
@@ -15,4 +17,4 @@ X-Plasma-API=declarativeappletscript
 X-Plasma-MainScript=ui/main.qml
 X-Plasma-RemoteLocation=
 X-KDE-PluginInfo-Category=Application Launchers
-Icon=folder-favorites
+Icon=folder
