diff --git a/package/contents/ui/FullRepresentation.qml b/package/contents/ui/FullRepresentation.qml
index e2223a1..00f1bcb 100644
--- a/package/contents/ui/FullRepresentation.qml
+++ b/package/contents/ui/FullRepresentation.qml
@@ -22,7 +22,7 @@ import org.kde.plasma.extras 2.0 as PlasmaExtras
 
 Item {
     property real mediumSpacing: 1.5*units.smallSpacing
-    property real textHeight: theme.defaultFont.pixelSize + theme.smallestFont.pixelSize + units.smallSpacing
+    property real textHeight: theme.defaultFont.pixelSize + units.smallSpacing
     property real itemHeight: Math.max(units.iconSizes.medium, textHeight)
 
     Layout.minimumWidth: widgetWidth
@@ -159,16 +159,6 @@ Item {
                                 width: 1
                                 height: units.smallSpacing
                             }
-                            PlasmaComponents.Label {
-                                text: model['url'].toString().replace('file://', '')
-                                font.pointSize: theme.smallestFont.pointSize
-                                opacity: isHovered ? 1.0 : 0.6
-                                width: parent.width
-                                height: theme.smallestFont.pixelSize
-                                elide: Text.ElideRight
-
-                                Behavior on opacity { NumberAnimation { duration: units.shortDuration * 3 } }
-                            }
                         }
                     }
 
