diff --git a/src/3rdparty/chromium/content/public/common/content_features.cc b/src/3rdparty/chromium/content/public/common/content_features.cc
index b7e601a77..c2c85d20f 100644
--- a/src/3rdparty/chromium/content/public/common/content_features.cc
+++ b/src/3rdparty/chromium/content/public/common/content_features.cc
@@ -172,12 +172,7 @@ const base::Feature kCooperativeScheduling{"CooperativeScheduling",
 // Enables crash reporting via Reporting API.
 // https://www.w3.org/TR/reporting/#crash-report
 const base::Feature kCrashReporting{"CrashReporting",
-#if BUILDFLAG(ENABLE_REPORTING)
-                                    base::FEATURE_ENABLED_BY_DEFAULT
-#else
-                                    base::FEATURE_DISABLED_BY_DEFAULT
-#endif
-};
+                                    base::FEATURE_DISABLED_BY_DEFAULT};
 
 // Puts save-data header in the holdback mode. This disables sending of
 // save-data header to origins, and to the renderer processes within Chrome.
