diff --git a/src/plugins/kdecorations/aurorae/src/aurorae.cpp b/src/plugins/kdecorations/aurorae/src/aurorae.cpp
index 1e91f3f..a799263 100644
--- a/src/plugins/kdecorations/aurorae/src/aurorae.cpp
+++ b/src/plugins/kdecorations/aurorae/src/aurorae.cpp
@@ -549,27 +549,7 @@ void Decoration::installTitleItem(QQuickItem *item)
 
 void Decoration::updateExtendedBorders()
 {
-    // extended sizes
-    const int extSize = settings()->largeSpacing();
-    int extLeft = m_extendedBorders->left();
-    int extRight = m_extendedBorders->right();
-    int extBottom = m_extendedBorders->bottom();
-
-    if (settings()->borderSize() == KDecoration2::BorderSize::None) {
-        if (!clientPointer()->isMaximizedHorizontally()) {
-            extLeft = qMax(m_extendedBorders->left(), extSize);
-            extRight = qMax(m_extendedBorders->right(), extSize);
-        }
-        if (!clientPointer()->isMaximizedVertically()) {
-            extBottom = qMax(m_extendedBorders->bottom(), extSize);
-        }
-
-    } else if (settings()->borderSize() == KDecoration2::BorderSize::NoSides && !clientPointer()->isMaximizedHorizontally()) {
-        extLeft = qMax(m_extendedBorders->left(), extSize);
-        extRight = qMax(m_extendedBorders->right(), extSize);
-    }
-
-    setResizeOnlyBorders(QMargins(extLeft, 0, extRight, extBottom));
+    setResizeOnlyBorders(QMargins(0, 0, 0, 0));
 }
 
 void Decoration::updateBlur()
