diff --git a/src/core_ghidra.cpp b/src/core_ghidra.cpp
index 5a2de68..9672fbd 100644
--- a/src/core_ghidra.cpp
+++ b/src/core_ghidra.cpp
@@ -44,7 +44,11 @@ struct ConfigVar
 
 		ut64 GetInt(RzConfig *cfg) const				{ return rz_config_get_i(cfg, name.c_str()); }
 		bool GetBool(RzConfig *cfg) const			{ return GetInt(cfg) != 0; }
-		std::string GetString(RzConfig *cfg) const	{ return rz_config_get(cfg, name.c_str()); }
+		std::string GetString(RzConfig *cfg) const	{
+			// returned pointer can be nullptr
+			auto val = rz_config_get(cfg, name.c_str());
+			return val ? std::string{val} : std::string{};
+		}
 
 		void Set(RzConfig *cfg, const char *s) const	{ rz_config_set(cfg, name.c_str(), s); }
 
