diff --git a/src/core/ksambashare.cpp b/src/core/ksambashare.cpp
index 9e315e4..85f10de 100644
--- a/src/core/ksambashare.cpp
+++ b/src/core/ksambashare.cpp
@@ -159,27 +159,8 @@ QString KSambaSharePrivate::testparmParamValue(const QString &parameterName)
 
         const int netbiosNameErrorIdx = errArray.indexOf("WARNING: The 'netbios name' is too long (max. 15 chars).");
         if (netbiosNameErrorIdx >= 0) {
-            // netbios name must be of at most 15 characters long
-            // means either netbios name is badly configured
-            // or not set and the default value is being used, it being "$(hostname)-W"
-            // which means any hostname longer than 13 characters will cause this warning
-            // when no netbios name was defined
-            // See https://www.novell.com/documentation/open-enterprise-server-2018/file_samba_cifs_lx/data/bc855e3.html
-            const QString defaultNetbiosName = QHostInfo::localHostName().append(QStringLiteral("-W"));
-            if (defaultNetbiosName.length() > 14) {
-                qCDebug(KIO_CORE) << "Your samba 'netbios name' parameter was longer than the authorized 15 characters.\n"
-                                  << "It may be because your hostname is longer than 13 and samba default 'netbios name' defaults to 'hostname-W', here:"
-                                  << defaultNetbiosName << "\n"
-                                  << "If that it is the case simply define a 'netbios name' parameter in /etc/samba/smb.conf at most 15 characters long";
-            } else {
-                qCDebug(KIO_CORE) << "Your samba 'netbios name' parameter was longer than the authorized 15 characters."
-                                  << "Please define a 'netbios name' parameter in /etc/samba/smb.conf at most 15 characters long";
-            }
             errArray.removeAt(netbiosNameErrorIdx);
         }
-        if (errArray.size() > 0) {
-            qCDebug(KIO_CORE) << "We got some errors while running testparm" << errArray.join("\n");
-        }
     }
 
     if (!stdOut.isEmpty()) {
@@ -210,12 +191,6 @@ QByteArray KSambaSharePrivate::getNetUserShareInfo()
             skipUserShare = true;
         } else if (stdErr.contains("usershares are currently disabled")) {
             skipUserShare = true;
-        } else {
-            // TODO: parse and process other error messages.
-            // create a parser for the error output and
-            // send error message somewhere
-            qCDebug(KIO_CORE) << "We got some errors while running 'net usershare info'";
-            qCDebug(KIO_CORE) << stdErr;
         }
     }
 
@@ -368,14 +343,6 @@ KSambaShareData::UserShareError KSambaSharePrivate::add(const KSambaShareData &s
     QByteArray stdOut;
     int ret = runProcess(QStringLiteral("net"), args, stdOut, m_stdErr);
 
-    // TODO: parse and process error messages.
-    if (!m_stdErr.isEmpty()) {
-        // create a parser for the error output and
-        // send error message somewhere
-        qCWarning(KIO_CORE) << "We got some errors while running 'net usershare add'" << args;
-        qCWarning(KIO_CORE) << m_stdErr;
-    }
-
     if (ret == 0 && !data.contains(shareData.name())) {
         // It needs to be added in this function explicitly, otherwise another instance of
         // KSambaShareDataPrivate will be created and added to data when the share
@@ -405,14 +372,6 @@ KSambaShareData::UserShareError KSambaSharePrivate::remove(const KSambaShareData
     QByteArray stdOut;
     int ret = runProcess(QStringLiteral("net"), args, stdOut, m_stdErr);
 
-    // TODO: parse and process error messages.
-    if (!m_stdErr.isEmpty()) {
-        // create a parser for the error output and
-        // send error message somewhere
-        qCWarning(KIO_CORE) << "We got some errors while running 'net usershare delete'" << args;
-        qCWarning(KIO_CORE) << m_stdErr;
-    }
-
     return (ret == 0) ? KSambaShareData::UserShareOk : KSambaShareData::UserShareSystemError;
 
     // NB: the share file gets deleted which leads us to reload and drop the ShareData, hence no explicit remove
@@ -457,9 +416,6 @@ QMap<QString, KSambaShareData> KSambaSharePrivate::parse(const QByteArray &users
                 shareData.dd->acl = value;
             } else if (key == QLatin1String("guest_ok")) {
                 shareData.dd->guestPermission = value;
-            } else {
-                qCWarning(KIO_CORE) << "Something nasty happen while parsing 'net usershare info'"
-                                    << "share:" << currentShare << "key:" << key;
             }
         } else if (line.trimmed().isEmpty()) {
             continue;
