diff --git a/src/core/ksambashare.cpp b/src/core/ksambashare.cpp
index e09d6cf..bbc974a 100644
--- a/src/core/ksambashare.cpp
+++ b/src/core/ksambashare.cpp
@@ -91,8 +91,6 @@ bool KSambaSharePrivate::findSmbConf()
         }
     }
 
-    qCDebug(KIO_CORE_SAMBASHARE) << "KSambaShare: Could not find smb.conf!";
-
     return false;
 }
 
@@ -148,8 +146,6 @@ QString KSambaSharePrivate::testparmParamValue(const QString &parameterName)
                 && err.at(0).startsWith("Load smb config files from")
                 && err.at(1).startsWith("Loaded services file OK.")) {
             //qDebug() << "Running testparm" << args;
-        } else {
-            qCWarning(KIO_CORE) << "We got some errors while running testparm" << stdErr;
         }
     }
 
@@ -181,12 +177,6 @@ QByteArray KSambaSharePrivate::getNetUserShareInfo()
             skipUserShare = true;
         } else if (stdErr.contains("usershares are currently disabled")) {
             skipUserShare = true;
-        } else {
-            //TODO: parse and process other error messages.
-            // create a parser for the error output and
-            // send error message somewhere
-            qCWarning(KIO_CORE) << "We got some errors while running 'net usershare info'";
-            qCWarning(KIO_CORE) << stdErr;
         }
     }
 
@@ -344,14 +334,6 @@ KSambaShareData::UserShareError KSambaSharePrivate::add(const KSambaShareData &s
 
     int ret = runProcess(QStringLiteral("net"), args, stdOut, stdErr);
 
-    //TODO: parse and process error messages.
-    if (!stdErr.isEmpty()) {
-        // create a parser for the error output and
-        // send error message somewhere
-        qCWarning(KIO_CORE) << "We got some errors while running 'net usershare add'" << args;
-        qCWarning(KIO_CORE) << stdErr;
-    }
-
     return (ret == 0) ? KSambaShareData::UserShareOk : KSambaShareData::UserShareSystemError;
 }
 
