diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6407690..c649494 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -143,7 +143,7 @@ add_subdirectory(src)
 # ccache executable
 #
 add_executable(ccache src/main.cpp)
-target_link_libraries(ccache PRIVATE standard_settings standard_warnings ccache_framework)
+target_link_libraries(ccache PRIVATE standard_settings standard_warnings ccache_framework -lpthread)
 
 #
 # Documentation
diff --git a/cmake/StdFilesystem.cmake b/cmake/StdFilesystem.cmake
index ca451ce..2696d0a 100644
--- a/cmake/StdFilesystem.cmake
+++ b/cmake/StdFilesystem.cmake
@@ -12,6 +12,7 @@ set(
     }
   ]=])
 
+set(CMAKE_REQUIRED_LIBRARIES pthread)
 check_cxx_source_compiles("${check_std_filesystem_source_code}" std_filesystem_without_libfs)
 
 if(NOT std_filesystem_without_libfs)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 79f0fd1..0d73b42 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -70,7 +70,7 @@ if(REDIS_STORAGE_BACKEND)
 endif()
 
 add_executable(test-lockfile test_lockfile.cpp)
-target_link_libraries(test-lockfile PRIVATE standard_settings standard_warnings ccache_framework)
+target_link_libraries(test-lockfile PRIVATE standard_settings standard_warnings ccache_framework -lpthread)
 
 add_subdirectory(core)
 add_subdirectory(storage)
