diff --git a/src/Button.cc b/src/Button.cc
index 1ec2cba..2e0c057 100644
--- a/src/Button.cc
+++ b/src/Button.cc
@@ -327,29 +327,6 @@ QColor Button::backgroundColor() const
         return Qt::transparent;
     }
 
-    //--- CloseButton
-    if (type() == KDecoration3::DecorationButtonType::Close) {
-        auto *decoratedClient = deco->window();
-        const QColor hoveredColor = decoratedClient->color(
-            KDecoration3::ColorGroup::Warning,
-            KDecoration3::ColorRole::Foreground
-        );
-        QColor normalColor = QColor(hoveredColor);
-        normalColor.setAlphaF(0);
-
-        if (isPressed()) {
-            const QColor pressedColor = decoratedClient->color(
-                KDecoration3::ColorGroup::Warning,
-                KDecoration3::ColorRole::Foreground
-            ).lighter();
-            return KColorUtils::mix(normalColor, pressedColor, m_transitionValue);
-        }
-
-        if (isHovered()) {
-            return KColorUtils::mix(normalColor, hoveredColor, m_transitionValue);
-        }
-    }
-
     //--- Checked
     if (isChecked() && type() != KDecoration3::DecorationButtonType::Maximize) {
         const QColor normalColor = deco->titleBarForegroundColor();
