diff --git a/Makefile.tool.am b/Makefile.tool.am
index 2bf90de..346bba6 100644
--- a/Makefile.tool.am
+++ b/Makefile.tool.am
@@ -17,7 +17,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_SEC_CAPS@ = \
 endif
 
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
 	$(TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@) $(TOOL_LDADD_COMMON)
diff --git a/cachegrind/Makefile.in b/cachegrind/Makefile.in
index 18bfcbc..bd875a6 100644
--- a/cachegrind/Makefile.in
+++ b/cachegrind/Makefile.in
@@ -672,7 +672,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/callgrind/Makefile.in b/callgrind/Makefile.in
index 8cf9272..72fb778 100644
--- a/callgrind/Makefile.in
+++ b/callgrind/Makefile.in
@@ -708,7 +708,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/dhat/Makefile.in b/dhat/Makefile.in
index eae1c0a..7bd7432 100644
--- a/dhat/Makefile.in
+++ b/dhat/Makefile.in
@@ -675,7 +675,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/drd/Makefile.in b/drd/Makefile.in
index a7dd4e5..3400e6d 100644
--- a/drd/Makefile.in
+++ b/drd/Makefile.in
@@ -767,7 +767,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/exp-bbv/Makefile.in b/exp-bbv/Makefile.in
index fdb2a69..9d8774e 100644
--- a/exp-bbv/Makefile.in
+++ b/exp-bbv/Makefile.in
@@ -621,7 +621,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/helgrind/Makefile.in b/helgrind/Makefile.in
index 1d069d3..abf01b5 100644
--- a/helgrind/Makefile.in
+++ b/helgrind/Makefile.in
@@ -709,7 +709,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/lackey/Makefile.in b/lackey/Makefile.in
index a6c4d99..b542abb 100644
--- a/lackey/Makefile.in
+++ b/lackey/Makefile.in
@@ -621,7 +621,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/massif/Makefile.in b/massif/Makefile.in
index e4d9888..a16dfcf 100644
--- a/massif/Makefile.in
+++ b/massif/Makefile.in
@@ -676,7 +676,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/memcheck/Makefile.in b/memcheck/Makefile.in
index 49713da..9800abf 100644
--- a/memcheck/Makefile.in
+++ b/memcheck/Makefile.in
@@ -709,7 +709,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/none/Makefile.in b/none/Makefile.in
index 432a991..aac7616 100644
--- a/none/Makefile.in
+++ b/none/Makefile.in
@@ -619,7 +619,7 @@ TOOL_DEPENDENCIES_@VGCONF_PLATFORM_PRI_CAPS@ = \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/coregrind/libcoregrind-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a \
 @VGCONF_HAVE_PLATFORM_SEC_TRUE@	$(top_builddir)/VEX/libvex-@VGCONF_ARCH_SEC@-@VGCONF_OS@.a
 
-TOOL_LDADD_COMMON = -lgcc \
+TOOL_LDADD_COMMON = \
 	$(top_builddir)/coregrind/libgcc-sup-@VGCONF_ARCH_PRI@-@VGCONF_OS@.a
 
 TOOL_LDADD_@VGCONF_PLATFORM_PRI_CAPS@ = \
diff --git a/valgrind.pc.in b/valgrind.pc.in
index 8a923cd..755cf99 100644
--- a/valgrind.pc.in
+++ b/valgrind.pc.in
@@ -11,6 +11,6 @@ Name: Valgrind
 Description: A dynamic binary instrumentation framework
 Version: @VERSION@
 Requires:
-Libs: -L${libdir}/valgrind -lcoregrind-@VGCONF_ARCH_PRI@-@VGCONF_OS@ -lvex-@VGCONF_ARCH_PRI@-@VGCONF_OS@ -lgcc
+Libs: -L${libdir}/valgrind -lcoregrind-@VGCONF_ARCH_PRI@-@VGCONF_OS@ -lvex-@VGCONF_ARCH_PRI@-@VGCONF_OS@
 Cflags: -I${includedir}
 
