diff --git a/coregrind/m_main.c b/coregrind/m_main.c
index c182fd9..686b13a 100644
--- a/coregrind/m_main.c
+++ b/coregrind/m_main.c
@@ -3266,7 +3266,7 @@ Addr VG_(get_initial_client_SP)( void )
      ___udivdi3
      ___umoddi3
 */
-#if defined(VGP_x86_darwin)
+#if defined(VGP_x86_linux)
 
 /* Routines for doing signed/unsigned 64 x 64 ==> 64 div and mod
    (udivdi3, umoddi3, divdi3, moddi3) using only 32 x 32 ==> 32
