diff --git a/Makefile.in b/Makefile.in
index 7e95b59..436b413 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -324,7 +324,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/configure b/configure
index 5a96128..7b6842e 100755
--- a/configure
+++ b/configure
@@ -13505,7 +13505,7 @@ fi
 
 fi
 
-VORBIS_LIBS="-lvorbis"
+VORBIS_LIBS="-lvorbis -logg"
 VORBISFILE_LIBS="-lvorbisfile"
 VORBISENC_LIBS="-lvorbisenc"
 
diff --git a/doc/Makefile.in b/doc/Makefile.in
index c0d29b7..413af1d 100644
--- a/doc/Makefile.in
+++ b/doc/Makefile.in
@@ -208,7 +208,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/examples/Makefile.in b/examples/Makefile.in
index fbaeaed..6dbb9f1 100644
--- a/examples/Makefile.in
+++ b/examples/Makefile.in
@@ -262,7 +262,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/include/Makefile.in b/include/Makefile.in
index 0ce7d24..d502585 100644
--- a/include/Makefile.in
+++ b/include/Makefile.in
@@ -268,7 +268,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/include/shout/Makefile.in b/include/shout/Makefile.in
index cfe5b6c..ad1c8a6 100644
--- a/include/shout/Makefile.in
+++ b/include/shout/Makefile.in
@@ -257,7 +257,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/m4/vorbis.m4 b/m4/vorbis.m4
index 17add29..675d420 100644
--- a/m4/vorbis.m4
+++ b/m4/vorbis.m4
@@ -38,7 +38,7 @@ if test "x$vorbis_prefix" != "x$ogg_prefix"; then
             ])
 fi
 
-VORBIS_LIBS="-lvorbis"
+VORBIS_LIBS="-lvorbis -logg"
 VORBISFILE_LIBS="-lvorbisfile"
 VORBISENC_LIBS="-lvorbisenc"
 
diff --git a/src/Makefile.in b/src/Makefile.in
index 09aac7f..f661027 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -353,7 +353,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/src/common/avl/Makefile.in b/src/common/avl/Makefile.in
index 0b400a6..e8f78b4 100644
--- a/src/common/avl/Makefile.in
+++ b/src/common/avl/Makefile.in
@@ -263,7 +263,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/src/common/httpp/Makefile.in b/src/common/httpp/Makefile.in
index 1e8f323..7c3b50f 100644
--- a/src/common/httpp/Makefile.in
+++ b/src/common/httpp/Makefile.in
@@ -265,7 +265,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/src/common/net/Makefile.in b/src/common/net/Makefile.in
index 512f5ea..656fe5f 100644
--- a/src/common/net/Makefile.in
+++ b/src/common/net/Makefile.in
@@ -264,7 +264,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/src/common/thread/Makefile.in b/src/common/thread/Makefile.in
index 263d6ac..5fdaa64 100644
--- a/src/common/thread/Makefile.in
+++ b/src/common/thread/Makefile.in
@@ -264,7 +264,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
diff --git a/src/common/timing/Makefile.in b/src/common/timing/Makefile.in
index 79842b5..7608a8e 100644
--- a/src/common/timing/Makefile.in
+++ b/src/common/timing/Makefile.in
@@ -264,7 +264,7 @@ THEORA_CFLAGS = @THEORA_CFLAGS@
 THEORA_LDFLAGS = @THEORA_LDFLAGS@
 THEORA_LIBS = @THEORA_LIBS@
 VERSION = @VERSION@
-VORBISENC_LIBS = @VORBISENC_LIBS@
+VORBISENC_LIBS = @VORBISENC_LIBS@ -logg
 VORBISFILE_LIBS = @VORBISFILE_LIBS@
 VORBIS_CFLAGS = @VORBIS_CFLAGS@
 VORBIS_LDFLAGS = @VORBIS_LDFLAGS@
