diff --git a/docs-xml/wscript_build b/docs-xml/wscript_build
index 86600ae..5885c16 100644
--- a/docs-xml/wscript_build
+++ b/docs-xml/wscript_build
@@ -27,7 +27,6 @@ manpages='''
          manpages/pdbedit.8
          manpages/profiles.1
          manpages/rpcclient.1
-         manpages/samba-regedit.8
          manpages/samba-tool.8
          manpages/samba.7
          manpages/samba.8
diff --git a/source3/utils/wscript_build b/source3/utils/wscript_build
index c6d1c29..ed8ed90 100644
--- a/source3/utils/wscript_build
+++ b/source3/utils/wscript_build
@@ -155,16 +155,6 @@ bld.SAMBA3_BINARY('split_tokens',
                  popt_samba3''',
                  install=False)
 
-bld.SAMBA3_BINARY('samba-regedit',
-                  source="""regedit.c regedit_samba3.c
-                            regedit_wrap.c regedit_treeview.c
-                            regedit_valuelist.c regedit_dialog.c
-                            regedit_hexedit.c regedit_list.c""",
-                  deps='''
-                  ncurses menu panel form registry smbconf popt_samba3_cmdline
-                  ''',
-                  enabled=bld.env.build_regedit)
-
 bld.SAMBA3_BINARY('testparm',
                  source='testparm.c',
                  deps='''
diff --git a/source3/wscript b/source3/wscript
index 6feda38..07728b1 100644
--- a/source3/wscript
+++ b/source3/wscript
@@ -62,8 +62,6 @@ def options(opt):
 
     opt.SAMBA3_ADD_OPTION('cluster-support', default=False)
 
-    opt.SAMBA3_ADD_OPTION('regedit', default=None)
-
     opt.SAMBA3_ADD_OPTION('fake-kaserver',
                           help=("Include AFS fake-kaserver support"), default=False)
 
@@ -1590,21 +1588,6 @@ main() {
             conf.DEFINE('HAVE_DBUS', '1')
 
     conf.env.build_regedit = False
-    if not Options.options.with_regedit == False:
-        conf.PROCESS_SEPARATE_RULE('system_ncurses')
-        if conf.CONFIG_SET('HAVE_NCURSES'):
-            conf.env.build_regedit = True
-
-    if conf.env.build_regedit:
-        Logs.info("building regedit")
-    else:
-        if Options.options.with_regedit == False:
-            Logs.info("not building regedit (--without-regedit)")
-        elif Options.options.with_regedit == True:
-            Logs.error("ncurses not available, cannot build regedit")
-            conf.fatal("ncurses not available, but --with-regedit was specified")
-        else:
-            Logs.info("ncurses not available, not building regedit")
 
     if conf.CHECK_HEADERS('ftw.h') and conf.CHECK_FUNCS('nftw'):
         conf.env.build_mvxattr = True
