diff --git a/src/aurorae.cpp b/src/aurorae.cpp
index 6eee013..919e832 100644
--- a/src/aurorae.cpp
+++ b/src/aurorae.cpp
@@ -528,27 +528,7 @@ void Decoration::installTitleItem(QQuickItem *item)
 
 void Decoration::updateExtendedBorders()
 {
-    // extended sizes
-    const int extSize = settings()->largeSpacing();
-    int extLeft = m_extendedBorders->left();
-    int extRight = m_extendedBorders->right();
-    int extBottom = m_extendedBorders->bottom();
-
-    if (settings()->borderSize() == KDecoration3::BorderSize::None) {
-        if (!window()->isMaximizedHorizontally()) {
-            extLeft = std::max(m_extendedBorders->left(), extSize);
-            extRight = std::max(m_extendedBorders->right(), extSize);
-        }
-        if (!window()->isMaximizedVertically()) {
-            extBottom = std::max(m_extendedBorders->bottom(), extSize);
-        }
-
-    } else if (settings()->borderSize() == KDecoration3::BorderSize::NoSides && !window()->isMaximizedHorizontally()) {
-        extLeft = std::max(m_extendedBorders->left(), extSize);
-        extRight = std::max(m_extendedBorders->right(), extSize);
-    }
-
-    setResizeOnlyBorders(QMargins(extLeft, 0, extRight, extBottom));
+    setResizeOnlyBorders(QMargins(0, 0, 0, 0));
 }
 
 void Decoration::updateBlur()
