diff --git a/src/.env.cpp.kate-swp b/src/.env.cpp.kate-swp
deleted file mode 100644
index 5fa71ad..0000000
Binary files a/src/.env.cpp.kate-swp and /dev/null differ
diff --git a/src/env.cpp b/src/env.cpp
index 7ed5426..3a030dd 100644
--- a/src/env.cpp
+++ b/src/env.cpp
@@ -85,7 +85,7 @@ static const electric_var_t electric_variables[] = {
     {L"SHLVL", electric_var_t::freadonly | electric_var_t::fexports},
     {L"history", electric_var_t::freadonly | electric_var_t::fcomputed},
     {L"pipestatus", electric_var_t::freadonly | electric_var_t::fcomputed},
-    {L"status", electric_var_t::freadonly | electric_var_t::fcomputed},
+    {L"status", electric_var_t::fcomputed},
     {L"version", electric_var_t::freadonly},
     {L"FISH_VERSION", electric_var_t::freadonly},
     {L"fish_pid", electric_var_t::freadonly},
