diff --git a/linphone-app/src/components/settings/SettingsModel.cpp b/linphone-app/src/components/settings/SettingsModel.cpp
index e6f727f..e14603f 100644
--- a/linphone-app/src/components/settings/SettingsModel.cpp
+++ b/linphone-app/src/components/settings/SettingsModel.cpp
@@ -451,7 +451,7 @@ void SettingsModel::setVideoDefinition (const QVariantMap &definition) {
 }
 
 bool SettingsModel::getVideoSupported () const {
-	return CoreManager::getInstance()->getCore()->videoSupported();
+	return false;
 }
 
 // -----------------------------------------------------------------------------
