diff --git a/linphone-app/ui/views/App/Main/Contacts.qml b/linphone-app/ui/views/App/Main/Contacts.qml
index ce4c70c..7222b09 100644
--- a/linphone-app/ui/views/App/Main/Contacts.qml
+++ b/linphone-app/ui/views/App/Main/Contacts.qml
@@ -48,15 +48,6 @@ ColumnLayout {
         onTextChanged: contacts.setFilter(text)
       }
 
-      ExclusiveButtons {
-        texts: [
-          qsTr('selectAllContacts'),
-          qsTr('selectConnectedContacts')
-        ]
-
-        onClicked: contacts.useConnectedFilter = !!button
-      }
-
       TextButtonB {
         text: qsTr('addContact')
         onClicked: window.setView('ContactEdit')
@@ -102,20 +93,6 @@ ColumnLayout {
 
             RowLayout {
               spacing: ContactsStyle.contact.spacing
-
-              PresenceLevel {
-                Layout.preferredHeight: ContactsStyle.contact.presenceLevelSize
-                Layout.preferredWidth: ContactsStyle.contact.presenceLevelSize
-                level: $contact.presenceLevel
-              }
-
-              Text {
-                Layout.fillWidth: true
-                color: ContactsStyle.contact.presence.color
-                elide: Text.ElideRight
-                font.pointSize: ContactsStyle.contact.presence.pointSize
-                text: Presence.getPresenceStatusAsString($contact.presenceStatus)
-              }
             }
           }
 
