diff --git a/linphone-app/ui/views/App/Main/Home.qml b/linphone-app/ui/views/App/Main/Home.qml
index 31d2487..d47b0a8 100644
--- a/linphone-app/ui/views/App/Main/Home.qml
+++ b/linphone-app/ui/views/App/Main/Home.qml
@@ -46,15 +46,6 @@ Rectangle {
       //   $title: qsTr('howToTitle')
       // }
 
-      ListElement {
-        $component: 'button'
-        $componentText: qsTr('inviteButton')
-        $description: qsTr('inviteDescription')
-        $view: 'InviteFriends'
-        $icon: 'home_invite_friends'
-        $title: qsTr('inviteTitle')
-      }
-
       ListElement {
         $component: 'button'
         $componentText: qsTr('assistantButton')
