diff --git a/src/backends/mysql/session.cpp b/src/backends/mysql/session.cpp
index 383248c..781b24a 100644
--- a/src/backends/mysql/session.cpp
+++ b/src/backends/mysql/session.cpp
@@ -355,7 +355,7 @@ mysql_session_backend::mysql_session_backend(
     }
     if (reconnect_p)
     {
-        my_bool reconnect = 1;
+        bool reconnect = 1;
         if (0 != mysql_options(conn_, MYSQL_OPT_RECONNECT, &reconnect))
         {
             clean_up();
