diff --git a/messageviewer/src/widgets/attachmentdialog.cpp b/messageviewer/src/widgets/attachmentdialog.cpp
index e9b4371..4c68b90 100644
--- a/messageviewer/src/widgets/attachmentdialog.cpp
+++ b/messageviewer/src/widgets/attachmentdialog.cpp
@@ -24,9 +24,7 @@ AttachmentDialog::AttachmentDialog(QWidget *parent, const QString &filenameText,
     , dialog(new QDialog(parent))
 {
     text = xi18nc("@info",
-                  "Open attachment <filename>%1</filename>?<nl/>"
-                  "Note that opening an attachment may compromise "
-                  "your system's security.",
+                  "Open attachment <filename>%1</filename>?",
                   filenameText);
 
     dialog->setWindowTitle(i18nc("@title:window", "Open Attachment?"));
