diff --git a/include/plugin_base.hpp b/include/plugin_base.hpp
index a2356fb..535bea9 100644
--- a/include/plugin_base.hpp
+++ b/include/plugin_base.hpp
@@ -1,6 +1,7 @@
 #ifndef PLUGIN_BASE_HPP
 #define PLUGIN_BASE_HPP
 
+#include <string>
 #include <gio/gio.h>
 #include <gst/gst.h>
 #include <sigc++/sigc++.h>
diff --git a/include/util.hpp b/include/util.hpp
index e1ba755..157bf2a 100644
--- a/include/util.hpp
+++ b/include/util.hpp
@@ -6,6 +6,8 @@
 #include <iostream>
 #include <thread>
 #include <vector>
+#include <array>
+#include <string>
 
 namespace util {
 
