diff --git a/makefile b/makefile
index ce71d0d..fe29dc8 100644
--- a/makefile
+++ b/makefile
@@ -4,16 +4,14 @@
 
 include $(dir $(lastword $(MAKEFILE_LIST)))build/bootstrap.make
 
-dirs := cutl tests
+dirs := cutl
 
 default  := $(out_base)/
-test     := $(out_base)/.test
 dist     := $(out_base)/.dist
 install  := $(out_base)/.install
 clean    := $(out_base)/.clean
 
 $(default): $(addprefix $(out_base)/,$(addsuffix /,$(dirs)))
-$(test): $(addprefix $(out_base)/,$(addsuffix /.test,$(dirs)))
 
 # No dist support for tests for now.
 dist_dirs := $(filter-out tests,$(dirs))
