diff --git a/src/views/dolphinview.cpp b/src/views/dolphinview.cpp
index f3bde88..af949e8 100644
--- a/src/views/dolphinview.cpp
+++ b/src/views/dolphinview.cpp
@@ -2387,39 +2387,7 @@ void DolphinView::updatePlaceholderLabel()
         return;
     }
 
-    if (m_loadingState == LoadingState::Canceled) {
-        m_placeholderLabel->setText(i18n("Loading canceled"));
-    } else if (!nameFilter().isEmpty()) {
-        m_placeholderLabel->setText(i18n("No items matching the filter"));
-    } else if (m_url.scheme() == QLatin1String("baloosearch") || m_url.scheme() == QLatin1String("filenamesearch")) {
-        m_placeholderLabel->setText(i18n("No items matching the search"));
-    } else if (m_url.scheme() == QLatin1String("trash") && m_url.path() == QLatin1String("/")) {
-        m_placeholderLabel->setText(i18n("Trash is empty"));
-    } else if (m_url.scheme() == QLatin1String("tags")) {
-        if (m_url.path() == QLatin1Char('/')) {
-            m_placeholderLabel->setText(i18n("No tags"));
-        } else {
-            const QString tagName = m_url.path().mid(1); // Remove leading /
-            m_placeholderLabel->setText(i18n("No files tagged with \"%1\"", tagName));
-        }
-
-    } else if (m_url.scheme() == QLatin1String("recentlyused")) {
-        m_placeholderLabel->setText(i18n("No recently used items"));
-    } else if (m_url.scheme() == QLatin1String("smb")) {
-        m_placeholderLabel->setText(i18n("No shared folders found"));
-    } else if (m_url.scheme() == QLatin1String("network")) {
-        m_placeholderLabel->setText(i18n("No relevant network resources found"));
-    } else if (m_url.scheme() == QLatin1String("mtp") && m_url.path() == QLatin1String("/")) {
-        m_placeholderLabel->setText(i18n("No MTP-compatible devices found"));
-    } else if (m_url.scheme() == QLatin1String("afc") && m_url.path() == QLatin1String("/")) {
-        m_placeholderLabel->setText(i18n("No Apple devices found"));
-    } else if (m_url.scheme() == QLatin1String("bluetooth")) {
-        m_placeholderLabel->setText(i18n("No Bluetooth devices found"));
-    } else {
-        m_placeholderLabel->setText(i18n("Folder is empty"));
-    }
-
-    m_placeholderLabel->setVisible(true);
+    m_placeholderLabel->setVisible(false);
 #ifndef QT_NO_ACCESSIBILITY
     if (QAccessible::isActive()) {
         static_cast<KItemListViewAccessible *>(QAccessible::queryAccessibleInterface(m_view))->announceNewlyLoadedLocation(m_placeholderLabel->text());
