diff --git a/src/views/dolphinview.cpp b/src/views/dolphinview.cpp
index 9a063d8..f666bfa 100644
--- a/src/views/dolphinview.cpp
+++ b/src/views/dolphinview.cpp
@@ -2114,27 +2114,5 @@ void DolphinView::updatePlaceholderLabel()
         return;
     }
 
-    if (!nameFilter().isEmpty()) {
-        m_placeholderLabel->setText(i18n("No items matching the filter"));
-    } else if (m_url.scheme() == QLatin1String("baloosearch") || m_url.scheme() == QLatin1String("filenamesearch")) {
-        m_placeholderLabel->setText(i18n("No items matching the search"));
-    } else if (m_url.scheme() == QLatin1String("trash") && m_url.path() == QLatin1String("/")) {
-        m_placeholderLabel->setText(i18n("Trash is empty"));
-    } else if (m_url.scheme() == QLatin1String("tags")) {
-        m_placeholderLabel->setText(i18n("No tags"));
-    } else if (m_url.scheme() == QLatin1String("recentlyused")) {
-        m_placeholderLabel->setText(i18n("No recently used items"));
-    } else if (m_url.scheme() == QLatin1String("smb")) {
-        m_placeholderLabel->setText(i18n("No shared folders found"));
-    } else if (m_url.scheme() == QLatin1String("network")) {
-        m_placeholderLabel->setText(i18n("No relevant network resources found"));
-    } else if (m_url.scheme() == QLatin1String("mtp") && m_url.path() == QLatin1String("/")) {
-        m_placeholderLabel->setText(i18n("No MTP-compatible devices found"));
-    } else if (m_url.scheme() == QLatin1String("bluetooth")) {
-        m_placeholderLabel->setText(i18n("No Bluetooth devices found"));
-    } else {
-        m_placeholderLabel->setText(i18n("Folder is empty"));
-    }
-
-    m_placeholderLabel->setVisible(true);
+    m_placeholderLabel->setVisible(false);
 }
