diff --git a/src/settings/viewmodes/viewsettingstab.cpp b/src/settings/viewmodes/viewsettingstab.cpp
index fc9e941..c5a4c52 100644
--- a/src/settings/viewmodes/viewsettingstab.cpp
+++ b/src/settings/viewmodes/viewsettingstab.cpp
@@ -77,6 +77,9 @@ ViewSettingsTab::ViewSettingsTab(Mode mode, QWidget *parent)
         m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "3"));
         m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "4"));
         m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "5"));
+        m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "6"));
+        m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "7"));
+        m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "8"));
         topLayout->addRow(i18nc("@label:listbox", "Maximum lines:"), m_maxLinesBox);
         break;
     }
