diff --git a/src/settings/viewmodes/viewsettingstab.cpp b/src/settings/viewmodes/viewsettingstab.cpp
index 6ac280e..d9bf5dd 100644
--- a/src/settings/viewmodes/viewsettingstab.cpp
+++ b/src/settings/viewmodes/viewsettingstab.cpp
@@ -85,6 +85,9 @@ ViewSettingsTab::ViewSettingsTab(Mode mode, QWidget* parent) :
         m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "3"));
         m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "4"));
         m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "5"));
+        m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "6"));
+        m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "7"));
+        m_maxLinesBox->addItem(i18nc("@item:inlistbox Maximum lines", "8"));
         topLayout->addRow(i18nc("@label:listbox", "Maximum lines:"), m_maxLinesBox);
         break;
     }
