diff --git a/src/player/PlayerComponent.cpp b/src/player/PlayerComponent.cpp
index 991bfe0..03ef2d0 100644
--- a/src/player/PlayerComponent.cpp
+++ b/src/player/PlayerComponent.cpp
@@ -1296,7 +1296,7 @@ void PlayerComponent::setVideoConfiguration()
   {
     hwdecMode = "auto-copy";
   }
-  mpv::qt::set_property(m_mpv, "hwdec", hwdecMode);
+  mpv::qt::set_property(m_mpv, "hwdec", "vaapi");
   mpv::qt::set_property(m_mpv, "hwdec-image-format", hwdecVTFormat);
 
   QVariant deinterlace = SettingsComponent::Get().value(SETTINGS_SECTION_VIDEO, "deinterlace");
