diff --git a/cargo_home/gentoo/libgit2-sys-0.14.2+1.5.1/build.rs b/cargo_home/gentoo/libgit2-sys-0.14.2+1.5.1/build.rs
index e888b94..ff81136 100644
--- a/cargo_home/gentoo/libgit2-sys-0.14.2+1.5.1/build.rs
+++ b/cargo_home/gentoo/libgit2-sys-0.14.2+1.5.1/build.rs
@@ -75,6 +75,7 @@ fn main() {
     // is some way with cc to have different instructions per-file?
     cfg.define("GIT_REGEX_BUILTIN", "1")
         .include("libgit2/deps/pcre")
+        .define("_GNU_SOURCE", Some("1"))
         .define("HAVE_STDINT_H", Some("1"))
         .define("HAVE_MEMMOVE", Some("1"))
         .define("NO_RECURSE", Some("1"))
