diff --git a/src/libs/utils/process_stub_unix.c b/src/libs/utils/process_stub_unix.c
index 376473d..e09790c 100644
--- a/src/libs/utils/process_stub_unix.c
+++ b/src/libs/utils/process_stub_unix.c
@@ -23,6 +23,9 @@
 **
 ****************************************************************************/
 
+#define _XOPEN_SOURCE
+#define __USE_XOPEN_EXTENDED
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
