diff --git a/srtcore/srt.h b/srtcore/srt.h
index 0a1e8fe..22210f6 100644
--- a/srtcore/srt.h
+++ b/srtcore/srt.h
@@ -74,15 +74,15 @@ written by
 // (out of these, only [[deprecated]] is supported in C++14, so
 // for all lesser standard use compiler-specific attributes).
 #if defined(__cplusplus) && __cplusplus > 201406
-#define SRT_ATR_UNUSED [[maybe_unused]]
-#define SRT_ATR_DEPRECATED [[deprecated]]
-#define SRT_ATR_NODISCARD [[nodiscard]]
+#define SRT_ATR_UNUSED
+#define SRT_ATR_DEPRECATED
+#define SRT_ATR_NODISCARD
 
 // GNUG is GNU C++; this syntax is also supported by Clang
 #elif defined( __GNUG__)
-#define SRT_ATR_UNUSED __attribute__((unused))
-#define SRT_ATR_DEPRECATED __attribute__((deprecated))
-#define SRT_ATR_NODISCARD __attribute__((warn_unused_result))
+#define SRT_ATR_UNUSED
+#define SRT_ATR_DEPRECATED
+#define SRT_ATR_NODISCARD
 #else
 #define SRT_ATR_UNUSED
 #define SRT_ATR_DEPRECATED
