diff --git a/ext/glslang/glslang/Include/PoolAlloc.h b/ext/glslang/glslang/Include/PoolAlloc.h
index 69bacb1..c215326 100644
--- a/ext/glslang/glslang/Include/PoolAlloc.h
+++ b/ext/glslang/glslang/Include/PoolAlloc.h
@@ -240,7 +240,7 @@ class TPoolAllocator {
 
     int numCalls;           // just an interesting statistic
     size_t totalBytes;      // just an interesting statistic
-private:
+public:
     TPoolAllocator& operator=(const TPoolAllocator&);  // don't allow assignment operator
     TPoolAllocator(const TPoolAllocator&);  // don't allow default copy constructor
 };
