diff --git a/CMakeLists.txt b/CMakeLists.txt
index 81361b7..b46dc18 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -280,7 +280,7 @@ set_target_properties(hwy PROPERTIES
 if(UNIX AND NOT APPLE)
   if(NOT HWY_EMSCRIPTEN)
     # For GCC __atomic_store_8, see #887
-    target_link_libraries(hwy atomic)
+    target_link_libraries(hwy)
   endif()
   # not supported by MSVC/Clang, safe to skip (we use DLLEXPORT annotations)
   set_property(TARGET hwy APPEND_STRING PROPERTY
