diff --git a/kdecoration/breezedecoration.cpp b/kdecoration/breezedecoration.cpp
index d648d08..3a3b413 100644
--- a/kdecoration/breezedecoration.cpp
+++ b/kdecoration/breezedecoration.cpp
@@ -464,22 +464,7 @@ void Decoration::recalculateBorders()
     setBorders(bordersFor(window()->nextScale()));
 
     // extended sizes
-    const qreal extSize = KDecoration3::snapToPixelGrid(settings()->largeSpacing(), window()->nextScale());
-    qreal extSides = 0;
-    qreal extBottom = 0;
-    if (hasNoBorders()) {
-        if (!isMaximizedHorizontally()) {
-            extSides = extSize;
-        }
-        if (!isMaximizedVertically()) {
-            extBottom = extSize;
-        }
-
-    } else if (hasNoSideBorders() && !isMaximizedHorizontally()) {
-        extSides = extSize;
-    }
-
-    setResizeOnlyBorders(QMarginsF(extSides, 0, extSides, extBottom));
+    setResizeOnlyBorders(QMargins(0, 0, 0, 0));
 }
 
 //________________________________________________________________
