diff --git a/kdecoration/breezedecoration.cpp b/kdecoration/breezedecoration.cpp
index 5e702fd..f3e5cbb 100644
--- a/kdecoration/breezedecoration.cpp
+++ b/kdecoration/breezedecoration.cpp
@@ -14,7 +14,6 @@
 #include "config/breezeconfigwidget.h"
 
 #include "breezebutton.h"
-#include "breezesizegrip.h"
 
 #include "breezeboxshadowrenderer.h"
 
@@ -159,9 +158,6 @@ namespace Breeze
             // last deco destroyed, clean up shadow
             g_sShadow.clear();
         }
-
-        deleteSizeGrip();
-
     }
 
     //________________________________________________________________
@@ -170,8 +166,6 @@ namespace Breeze
         if( m_opacity == value ) return;
         m_opacity = value;
         update();
-
-        if( m_sizeGrip ) m_sizeGrip->update();
     }
 
     //________________________________________________________________
@@ -327,9 +321,6 @@ namespace Breeze
     //________________________________________________________________
     void Decoration::updateSizeGripVisibility()
     {
-        const auto c = client().toStrongRef();
-        if( m_sizeGrip )
-        { m_sizeGrip->setVisible( c->isResizeable() && !isMaximized() && !c->isShaded() ); }
     }
 
     //________________________________________________________________
@@ -397,10 +388,6 @@ namespace Breeze
         // shadow
         updateShadow();
 
-        // size grip
-        if( hasNoBorders() && m_internalSettings->drawSizeGrip() ) createSizeGrip();
-        else deleteSizeGrip();
-
     }
 
     //________________________________________________________________
@@ -434,21 +421,7 @@ namespace Breeze
         setBorders(QMargins(left, top, right, bottom));
 
         // extended sizes
-        const int extSize = s->largeSpacing();
-        int extSides = 0;
-        int extBottom = 0;
-        if( hasNoBorders() )
-        {
-            if( !isMaximizedHorizontally() ) extSides = extSize;
-            if( !isMaximizedVertically() ) extBottom = extSize;
-
-        } else if( hasNoSideBorders() && !isMaximizedHorizontally() ) {
-
-            extSides = extSize;
-
-        }
-
-        setResizeOnlyBorders(QMargins(extSides, 0, extSides, extBottom));
+        setResizeOnlyBorders(QMargins(0, 0, 0, 0));
     }
 
     //________________________________________________________________
@@ -840,36 +813,11 @@ namespace Breeze
     //_________________________________________________________________
     void Decoration::createSizeGrip()
     {
-
-        // do nothing if size grip already exist
-        if( m_sizeGrip ) return;
-
-        #if BREEZE_HAVE_X11
-        if( !QX11Info::isPlatformX11() ) return;
-
-        // access client
-        auto c = client().toStrongRef();
-        if( !c ) return;
-
-        if( c->windowId() != 0 )
-        {
-            m_sizeGrip = new SizeGrip( this );
-            connect( c.data(), &KDecoration2::DecoratedClient::maximizedChanged, this, &Decoration::updateSizeGripVisibility );
-            connect( c.data(), &KDecoration2::DecoratedClient::shadedChanged, this, &Decoration::updateSizeGripVisibility );
-            connect( c.data(), &KDecoration2::DecoratedClient::resizeableChanged, this, &Decoration::updateSizeGripVisibility );
-        }
-        #endif
-
     }
 
     //_________________________________________________________________
     void Decoration::deleteSizeGrip()
     {
-        if( m_sizeGrip )
-        {
-            m_sizeGrip->deleteLater();
-            m_sizeGrip = nullptr;
-        }
     }
     
     void Decoration::setScaledCornerRadius()
diff --git a/kdecoration/breezedecoration.h b/kdecoration/breezedecoration.h
index 9a55b2b..200be6f 100644
--- a/kdecoration/breezedecoration.h
+++ b/kdecoration/breezedecoration.h
@@ -123,16 +123,13 @@ namespace Breeze
         void createSizeGrip();
         void deleteSizeGrip();
         SizeGrip* sizeGrip() const
-        { return m_sizeGrip; }
+        { return nullptr; }
         //@}
 
         InternalSettingsPtr m_internalSettings;
         KDecoration2::DecorationButtonGroup *m_leftButtons = nullptr;
         KDecoration2::DecorationButtonGroup *m_rightButtons = nullptr;
 
-        //* size grip widget
-        SizeGrip *m_sizeGrip = nullptr;
-
         //* active state change animation
         QVariantAnimation *m_animation;
         QVariantAnimation *m_shadowAnimation;
diff --git a/kdecoration/breezesizegrip.h b/kdecoration/breezesizegrip.h
index ff24046..41acab7 100644
--- a/kdecoration/breezesizegrip.h
+++ b/kdecoration/breezesizegrip.h
@@ -68,7 +68,7 @@ namespace Breeze
         //* grip size
         enum {
             Offset = 0,
-            GripSize = 14,
+            GripSize = 0,
         };
 
         //* decoration
