diff --git a/kdecoration/breezedecoration.cpp b/kdecoration/breezedecoration.cpp
index cc5bb75..522009a 100644
--- a/kdecoration/breezedecoration.cpp
+++ b/kdecoration/breezedecoration.cpp
@@ -428,22 +428,7 @@ void Decoration::recalculateBorders()
     setBorders(QMargins(left, top, right, bottom));
 
     // extended sizes
-    const int extSize = s->largeSpacing();
-    int extSides = 0;
-    int extBottom = 0;
-    if (hasNoBorders()) {
-        if (!isMaximizedHorizontally()) {
-            extSides = extSize;
-        }
-        if (!isMaximizedVertically()) {
-            extBottom = extSize;
-        }
-
-    } else if (hasNoSideBorders() && !isMaximizedHorizontally()) {
-        extSides = extSize;
-    }
-
-    setResizeOnlyBorders(QMargins(extSides, 0, extSides, extBottom));
+    setResizeOnlyBorders(QMargins(0, 0, 0, 0));
 
     // Update shadows and clear outline to make sure outline changes when borders are changed
     updateShadow();
