diff --git a/libnotificationmanager/job_p.cpp b/libnotificationmanager/job_p.cpp
index 2824d48..f51663d 100644
--- a/libnotificationmanager/job_p.cpp
+++ b/libnotificationmanager/job_p.cpp
@@ -218,17 +218,8 @@ void JobPrivate::finish()
     // Unregister the dbus service since the client is done with it
     QDBusConnection::sessionBus().unregisterObject(m_objectPath.path());
 
-    // When user canceled transfer, remove it without notice
-    if (m_error == KIO::ERR_USER_CANCELED) {
-        emit closed();
-        return;
-    }
-
-    Job *job = static_cast<Job *>(parent());
-    // update timestamp
-    job->resetUpdated();
-    // when it was hidden in history, bring it up again
-    job->setDismissed(false);
+    // Remove job item from history when done
+    emit closed();
 }
 
 // JobViewV2
