diff --git a/libnotificationmanager/notification.cpp b/libnotificationmanager/notification.cpp
index ca7cc6a..eb626a5 100644
--- a/libnotificationmanager/notification.cpp
+++ b/libnotificationmanager/notification.cpp
@@ -391,22 +391,7 @@ void Notification::Private::processHints(const QVariantMap &hints)
     eventId = hints.value(QStringLiteral("x-kde-eventId")).toString();
     xdgTokenAppId = hints.value(QStringLiteral("x-kde-xdgTokenAppId")).toString();
 
-    bool ok;
-    const int urgency = hints.value(QStringLiteral("urgency")).toInt(&ok); // DBus type is actually "byte"
-    if (ok) {
-        // FIXME use separate enum again
-        switch (urgency) {
-        case 0:
-            setUrgency(Notifications::LowUrgency);
-            break;
-        case 1:
-            setUrgency(Notifications::NormalUrgency);
-            break;
-        case 2:
-            setUrgency(Notifications::CriticalUrgency);
-            break;
-        }
-    }
+    setUrgency(Notifications::LowUrgency);
 
     resident = hints.value(QStringLiteral("resident")).toBool();
     transient = hints.value(QStringLiteral("transient")).toBool();
