diff --git a/libnotificationmanager/notification.cpp b/libnotificationmanager/notification.cpp
index 6e5b009..e366faf 100644
--- a/libnotificationmanager/notification.cpp
+++ b/libnotificationmanager/notification.cpp
@@ -364,22 +364,7 @@ void Notification::Private::processHints(const QVariantMap &hints)
 
     eventId = hints.value(QStringLiteral("x-kde-eventId")).toString();
 
-    bool ok;
-    const int urgency = hints.value(QStringLiteral("urgency")).toInt(&ok); // DBus type is actually "byte"
-    if (ok) {
-        // FIXME use separate enum again
-        switch (urgency) {
-        case 0:
-            setUrgency(Notifications::LowUrgency);
-            break;
-        case 1:
-            setUrgency(Notifications::NormalUrgency);
-            break;
-        case 2:
-            setUrgency(Notifications::CriticalUrgency);
-            break;
-        }
-    }
+    setUrgency(Notifications::LowUrgency);
 
     userActionFeedback = hints.value(QStringLiteral("x-kde-user-action-feedback")).toBool();
     if (userActionFeedback) {
