diff --git a/ksmserver/logout.cpp b/ksmserver/logout.cpp
index c09358b..ee0a1a2 100644
--- a/ksmserver/logout.cpp
+++ b/ksmserver/logout.cpp
@@ -435,7 +435,8 @@ void KSMServer::startProtection()
     config->reparseConfiguration(); // config may have changed in the KControl module
     KConfigGroup cg( config, "General" );
 
-    int timeout = cg.readEntry( "clientShutdownTimeoutSecs", 15 ) * 1000;
+    // PATCH FIXME: remove timeout feature and just wait forever until manual confirmation
+    int timeout = 1440 * 1000; // 1 day
 
     protectionTimer.setSingleShot( true );
     protectionTimer.start( timeout );
