diff --git a/applets/mediacontroller/contents/ui/ExpandedRepresentation.qml b/applets/mediacontroller/contents/ui/ExpandedRepresentation.qml
index 2f89d33..b837714 100644
--- a/applets/mediacontroller/contents/ui/ExpandedRepresentation.qml
+++ b/applets/mediacontroller/contents/ui/ExpandedRepresentation.qml
@@ -42,7 +42,7 @@ Item {
     property double position: mpris2Source.currentData.Position || 0
     readonly property real rate: mpris2Source.currentData.Rate || 1
     readonly property double length: currentMetadata ? currentMetadata["mpris:length"] || 0 : 0
-    readonly property bool canSeek: mpris2Source.currentData.CanSeek || false
+    readonly property bool canSeek: false // mpris2Source.currentData.CanSeek || false
     readonly property bool softwareRendering: GraphicsInfo.api === GraphicsInfo.Software
 
     // only show hours (the default for KFormat) when track is actually longer than an hour
@@ -313,7 +313,7 @@ Item {
                 }
             }
         }
-
+/*
         Item {
             implicitHeight: units.smallSpacing
         }
@@ -483,6 +483,7 @@ Item {
                 disablePositionUpdate = false
             }
         }
+*/
     }
 
     Timer {
diff --git a/applets/mediacontroller/contents/ui/main.qml b/applets/mediacontroller/contents/ui/main.qml
index 23ad8f7..77ef2bc 100644
--- a/applets/mediacontroller/contents/ui/main.qml
+++ b/applets/mediacontroller/contents/ui/main.qml
@@ -72,7 +72,7 @@ Item {
 
     property var mprisSourcesModel: []
 
-    readonly property bool canControl: (!root.noPlayer && mpris2Source.currentData.CanControl) || false
+    readonly property bool canControl: false // (!root.noPlayer && mpris2Source.currentData.CanControl) || false
     readonly property bool canGoPrevious: (canControl && mpris2Source.currentData.CanGoPrevious) || false
     readonly property bool canGoNext: (canControl && mpris2Source.currentData.CanGoNext) || false
     readonly property bool canPlay: (canControl && mpris2Source.currentData.CanPlay) || false
