diff --git a/shell/panelshadows.cpp b/shell/panelshadows.cpp
index fa5c770..f905454 100644
--- a/shell/panelshadows.cpp
+++ b/shell/panelshadows.cpp
@@ -517,6 +517,7 @@ void PanelShadows::Private::freeWaylandBuffers()
 
 void PanelShadows::Private::updateShadow(const QWindow *window, Plasma::FrameSvg::EnabledBorders enabledBorders)
 {
+#if 0
     if (!hasShadows()) {
         return;
     }
@@ -529,11 +530,12 @@ void PanelShadows::Private::updateShadow(const QWindow *window, Plasma::FrameSvg
     if (m_wayland.manager) {
         updateShadowWayland(window, enabledBorders);
     }
+#endif
 }
 
 void PanelShadows::Private::updateShadowX11(const QWindow *window, Plasma::FrameSvg::EnabledBorders enabledBorders)
 {
-#if HAVE_X11
+#if 0
     if (m_shadowPixmaps.isEmpty()) {
         setupPixmaps();
     }
@@ -553,6 +555,7 @@ void PanelShadows::Private::updateShadowX11(const QWindow *window, Plasma::Frame
 
 void PanelShadows::Private::updateShadowWayland(const QWindow *window, Plasma::FrameSvg::EnabledBorders enabledBorders)
 {
+#if 0
     if (!m_wayland.shmPool) {
         return;
     }
@@ -651,6 +654,7 @@ void PanelShadows::Private::updateShadowWayland(const QWindow *window, Plasma::F
     shadow->setOffsets(margins);
     shadow->commit();
     surface->commit(KWayland::Client::Surface::CommitFlag::None);
+#endif
 }
 
 void PanelShadows::Private::clearShadow(const QWindow *window)
