diff --git a/shell/panelshadows.cpp b/shell/panelshadows.cpp
index 5cec447..da5be35 100644
--- a/shell/panelshadows.cpp
+++ b/shell/panelshadows.cpp
@@ -112,24 +112,6 @@ void PanelShadows::setEnabledBorders(QWindow *window, Plasma::FrameSvg::EnabledB
 
 void PanelShadows::Private::updateShadows()
 {
-    const bool hadShadowsBefore = !m_tiles.isEmpty();
-
-    // has shadows now?
-    if (hasShadows()) {
-        if (hadShadowsBefore) {
-            clearTiles();
-        }
-        for (auto i = m_windows.constBegin(); i != m_windows.constEnd(); ++i) {
-            updateShadow(i.key(), i.value());
-        }
-    } else {
-        if (hadShadowsBefore) {
-            for (auto i = m_windows.constBegin(); i != m_windows.constEnd(); ++i) {
-                clearShadow(i.key());
-            }
-            clearTiles();
-        }
-    }
 }
 
 void PanelShadows::Private::initTile(const QString &element)
@@ -145,15 +127,6 @@ void PanelShadows::Private::initTile(const QString &element)
 void PanelShadows::Private::setupTiles()
 {
     clearTiles();
-
-    initTile(QStringLiteral("shadow-top"));
-    initTile(QStringLiteral("shadow-topright"));
-    initTile(QStringLiteral("shadow-right"));
-    initTile(QStringLiteral("shadow-bottomright"));
-    initTile(QStringLiteral("shadow-bottom"));
-    initTile(QStringLiteral("shadow-bottomleft"));
-    initTile(QStringLiteral("shadow-left"));
-    initTile(QStringLiteral("shadow-topleft"));
 }
 
 void PanelShadows::Private::clearTiles()
@@ -163,116 +136,6 @@ void PanelShadows::Private::clearTiles()
 
 void PanelShadows::Private::updateShadow(QWindow *window, Plasma::FrameSvg::EnabledBorders enabledBorders)
 {
-    if (!hasShadows()) {
-        return;
-    }
-
-    if (m_tiles.isEmpty()) {
-        setupTiles();
-    }
-
-    KWindowShadow *&shadow = m_shadows[window];
-
-    if (!shadow) {
-        shadow = new KWindowShadow(q);
-    }
-
-    if (shadow->isCreated()) {
-        shadow->destroy();
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::TopBorder) {
-        shadow->setTopTile(m_tiles.at(0));
-    } else {
-        shadow->setTopTile(nullptr);
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::TopBorder && enabledBorders & Plasma::FrameSvg::RightBorder) {
-        shadow->setTopRightTile(m_tiles.at(1));
-    } else {
-        shadow->setTopRightTile(nullptr);
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::RightBorder) {
-        shadow->setRightTile(m_tiles.at(2));
-    } else {
-        shadow->setRightTile(nullptr);
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::BottomBorder && enabledBorders & Plasma::FrameSvg::RightBorder) {
-        shadow->setBottomRightTile(m_tiles.at(3));
-    } else {
-        shadow->setBottomRightTile(nullptr);
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::BottomBorder) {
-        shadow->setBottomTile(m_tiles.at(4));
-    } else {
-        shadow->setBottomTile(nullptr);
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::BottomBorder && enabledBorders & Plasma::FrameSvg::LeftBorder) {
-        shadow->setBottomLeftTile(m_tiles.at(5));
-    } else {
-        shadow->setBottomLeftTile(nullptr);
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::LeftBorder) {
-        shadow->setLeftTile(m_tiles.at(6));
-    } else {
-        shadow->setLeftTile(nullptr);
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::TopBorder && enabledBorders & Plasma::FrameSvg::LeftBorder) {
-        shadow->setTopLeftTile(m_tiles.at(7));
-    } else {
-        shadow->setTopLeftTile(nullptr);
-    }
-
-    QMargins padding;
-
-    if (enabledBorders & Plasma::FrameSvg::TopBorder) {
-        const QSize marginHint = q->elementSize(QStringLiteral("shadow-hint-top-margin"));
-        if (marginHint.isValid()) {
-            padding.setTop(marginHint.height());
-        } else {
-            padding.setTop(m_tiles[0]->image().height());
-        }
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::RightBorder) {
-        const QSize marginHint = q->elementSize(QStringLiteral("shadow-hint-right-margin"));
-        if (marginHint.isValid()) {
-            padding.setRight(marginHint.width());
-        } else {
-            padding.setRight(m_tiles[2]->image().width());
-        }
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::BottomBorder) {
-        const QSize marginHint = q->elementSize(QStringLiteral("shadow-hint-bottom-margin"));
-        if (marginHint.isValid()) {
-            padding.setBottom(marginHint.height());
-        } else {
-            padding.setBottom(m_tiles[4]->image().height());
-        }
-    }
-
-    if (enabledBorders & Plasma::FrameSvg::LeftBorder) {
-        const QSize marginHint = q->elementSize(QStringLiteral("shadow-hint-left-margin"));
-        if (marginHint.isValid()) {
-            padding.setLeft(marginHint.width());
-        } else {
-            padding.setLeft(m_tiles[6]->image().width());
-        }
-    }
-
-    shadow->setPadding(padding);
-    shadow->setWindow(window);
-
-    if (!shadow->create()) {
-        qCWarning(PLASMASHELL) << "Couldn't create KWindowShadow for" << window;
-    }
 }
 
 void PanelShadows::Private::clearShadow(QWindow *window)
