diff --git a/runners/services/servicerunner.cpp b/runners/services/servicerunner.cpp
index 32d7b48..0851109 100644
--- a/runners/services/servicerunner.cpp
+++ b/runners/services/servicerunner.cpp
@@ -296,11 +296,7 @@ private:
             }
 
             if (isKCM) {
-                if (service->parentApp() == QStringLiteral("kinfocenter")) {
-                    match.setMatchCategory(i18n("System Information"));
-                } else {
-                    match.setMatchCategory(i18n("System Settings"));
-                }
+                match.setMatchCategory(i18n("System Settings"));
                 // KCMs are, on the balance, less relevant. Drop it ever so much. So they may get outscored
                 // by an otherwise equally applicable match.
                 relevance -= .001;
@@ -466,18 +462,6 @@ void ServiceRunner::run(const Plasma::RunnerContext &context, const Plasma::Quer
 
     const QString actionName = QUrlQuery(dataUrl).queryItemValue(QStringLiteral("action"));
     if (actionName.isEmpty()) {
-        // We want to load kcms directly with systemsettings,
-        // but we can't completely replace kcmshell with systemsettings
-        // as we need to be able to load kcms without plasma and we can't
-        // implement all kcmshell features into systemsettings
-        if (service->serviceTypes().contains(QLatin1String("KCModule"))) {
-            if (service->parentApp() == QStringLiteral("kinfocenter")) {
-                service->setExec(QStringLiteral("kinfocenter ") + service->desktopEntryName());
-            // We can't display a KCM in systemsettings if it has no parent, BUG: 423612
-            } else if (!service->property("X-KDE-System-Settings-Parent-Category").toString().isEmpty()) {
-                service->setExec(QStringLiteral("systemsettings5 ") + service->desktopEntryName());
-            }
-        }
         job = new KIO::ApplicationLauncherJob(service);
     } else {
         const auto actions = service->actions();
