diff --git a/src/corelib/io/qfilesystemwatcher.cpp b/src/corelib/io/qfilesystemwatcher.cpp
index 54460aff..b335dbac 100644
--- a/src/corelib/io/qfilesystemwatcher.cpp
+++ b/src/corelib/io/qfilesystemwatcher.cpp
@@ -307,7 +307,6 @@ QFileSystemWatcher::~QFileSystemWatcher()
 bool QFileSystemWatcher::addPath(const QString &path)
 {
     if (path.isEmpty()) {
-        qWarning("QFileSystemWatcher::addPath: path is empty");
         return true;
     }
 
@@ -356,7 +355,6 @@ QStringList QFileSystemWatcher::addPaths(const QStringList &paths)
     QStringList p = empty_paths_pruned(paths);
 
     if (p.isEmpty()) {
-        qWarning("QFileSystemWatcher::addPaths: list is empty");
         return p;
     }
     qCDebug(lcWatcher) << "adding" << paths;
@@ -406,7 +404,6 @@ QStringList QFileSystemWatcher::addPaths(const QStringList &paths)
 bool QFileSystemWatcher::removePath(const QString &path)
 {
     if (path.isEmpty()) {
-        qWarning("QFileSystemWatcher::removePath: path is empty");
         return true;
     }
 
@@ -432,7 +429,6 @@ QStringList QFileSystemWatcher::removePaths(const QStringList &paths)
     QStringList p = empty_paths_pruned(paths);
 
     if (p.isEmpty()) {
-        qWarning("QFileSystemWatcher::removePaths: list is empty");
         return p;
     }
     qCDebug(lcWatcher) << "removing" << paths;
