diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 389cf5a..ebe71e4 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -118,12 +118,12 @@ if  (${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
         add_definitions(-Wimplicit-fallthrough -Wshorten-64-to-32)
     endif()
     if (${CMAKE_CXX_COMPILER} MATCHES  ".*mingw.*")
-        target_link_libraries(graphite2 kernel32 msvcr90 mingw32 gcc user32)
+        target_link_libraries(graphite2 kernel32 msvcr90 mingw32 user32)
     else (${CMAKE_CXX_COMPILER} MATCHES  ".*mingw.*")
         if (GRAPHITE2_SANITIZERS)
-            target_link_libraries(graphite2 c gcc_s)
+            target_link_libraries(graphite2 c)
         else ()
-            target_link_libraries(graphite2 c gcc)
+            target_link_libraries(graphite2 c)
         endif ()
     endif()
     include(Graphite)
